{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "39f38bd8",
   "metadata": {},
   "source": [
    "# PROCESSING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d112d79b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "c1d6a598",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_df = pd.read_csv('data/gene_list.txt',header=None)\n",
    "gene_list = gene_df[0].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "49ef012a",
   "metadata": {},
   "outputs": [],
   "source": [
    "clin_patient = pd.read_csv('data/data_clinical_patient.txt', sep='\\t', comment='#')\n",
    "clin_sample = pd.read_csv('data/data_clinical_sample.txt', sep='\\t', comment='#')\n",
    "mut = pd.read_csv('data/data_mutations.txt', sep='\\t', comment='#')\n",
    "cna = pd.read_csv('data/data_cna.txt', sep='\\t')\n",
    "rna = pd.read_csv('data/data_mrna_illumina_microarray_zscores_ref_diploid_samples.txt', sep='\\t', comment='#')\n",
    "meth = pd.read_csv('data/data_methylation_promoters_rrbs.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49a6498d",
   "metadata": {},
   "source": [
    "## mut "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "37ed36b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of genes:  173\n",
      "number of patients:  2369\n"
     ]
    }
   ],
   "source": [
    "print('number of genes: ',len(mut.Hugo_Symbol.unique()))\n",
    "print('number of patients: ',len(mut.Tumor_Sample_Barcode.unique()))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c042402",
   "metadata": {},
   "source": [
    "Select mutations involving genes that are present in the breast cancer gene list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "de06e115",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of mutated genes that are present in breast cancer gene list:  28\n",
      "Number of patients with small mutations in the selected genes:  2135\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TP53</td>\n",
       "      <td>MTS-T0058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TP53</td>\n",
       "      <td>MTS-T0059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ERBB2</td>\n",
       "      <td>MTS-T0059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>EP300</td>\n",
       "      <td>MTS-T0059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>MTS-T0061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17257</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>MB-0899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17261</th>\n",
       "      <td>TP53</td>\n",
       "      <td>MB-0901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17265</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>MB-0904</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17267</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>MB-0906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17269</th>\n",
       "      <td>TP53</td>\n",
       "      <td>MB-0906</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4420 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Hugo_Symbol Tumor_Sample_Barcode\n",
       "0            TP53            MTS-T0058\n",
       "10           TP53            MTS-T0059\n",
       "11          ERBB2            MTS-T0059\n",
       "14          EP300            MTS-T0059\n",
       "23         PIK3CA            MTS-T0061\n",
       "...           ...                  ...\n",
       "17257      PIK3CA              MB-0899\n",
       "17261        TP53              MB-0901\n",
       "17265      PIK3CA              MB-0904\n",
       "17267      PIK3CA              MB-0906\n",
       "17269        TP53              MB-0906\n",
       "\n",
       "[4420 rows x 2 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_filter = mut[['Hugo_Symbol','Tumor_Sample_Barcode']].copy()\n",
    "mut_filter = mut_filter[mut_filter['Hugo_Symbol'].isin(gene_list)]\n",
    "mut_filter = mut_filter.drop_duplicates()\n",
    "print('Number of mutated genes that are present in breast cancer gene list: ',len(mut_filter.Hugo_Symbol.unique()))\n",
    "print('Number of patients with small mutations in the selected genes: ',len(mut_filter.Tumor_Sample_Barcode.unique()))\n",
    "mut_filter"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d74c6723",
   "metadata": {},
   "source": [
    "From the filtered table, create a new binary-valued matrix with patients as rows and genes as columns. Entries are 0 if no mutation is present in the selected gene for the corresponding patient, and 1 if mutations are present. No difference is reported if more than one mutation is present in the same gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "228171f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AKT1</th>\n",
       "      <th>ARID1A</th>\n",
       "      <th>ARID1B</th>\n",
       "      <th>BAP1</th>\n",
       "      <th>BRCA1</th>\n",
       "      <th>BRCA2</th>\n",
       "      <th>BRIP1</th>\n",
       "      <th>CASP8</th>\n",
       "      <th>CDH1</th>\n",
       "      <th>CDKN1B</th>\n",
       "      <th>...</th>\n",
       "      <th>NCOR1</th>\n",
       "      <th>NOTCH1</th>\n",
       "      <th>PBRM1</th>\n",
       "      <th>PIK3CA</th>\n",
       "      <th>PTEN</th>\n",
       "      <th>RB1</th>\n",
       "      <th>SMARCD1</th>\n",
       "      <th>STK11</th>\n",
       "      <th>TBX3</th>\n",
       "      <th>TP53</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0002</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0005</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0006</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0008</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0010</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2428</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2429</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2430</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2431</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2432</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2135 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           AKT1  ARID1A  ARID1B  BAP1  BRCA1  BRCA2  BRIP1  CASP8  CDH1  \\\n",
       "MB-0002       0       0       0     0      0      0      0      0     0   \n",
       "MB-0005       0       0       0     0      0      0      0      0     0   \n",
       "MB-0006       0       0       0     0      0      0      0      0     0   \n",
       "MB-0008       0       0       0     0      0      0      0      0     0   \n",
       "MB-0010       0       0       0     0      0      0      0      0     0   \n",
       "...         ...     ...     ...   ...    ...    ...    ...    ...   ...   \n",
       "MTS-T2428     0       0       0     0      0      0      0      0     0   \n",
       "MTS-T2429     0       0       0     0      0      0      0      0     1   \n",
       "MTS-T2430     0       0       0     0      0      0      0      0     0   \n",
       "MTS-T2431     0       0       0     0      0      0      0      0     0   \n",
       "MTS-T2432     0       0       0     0      0      0      0      0     0   \n",
       "\n",
       "           CDKN1B  ...  NCOR1  NOTCH1  PBRM1  PIK3CA  PTEN  RB1  SMARCD1  \\\n",
       "MB-0002         0  ...      0       0      0       0     0    0        0   \n",
       "MB-0005         0  ...      1       0      0       1     0    0        0   \n",
       "MB-0006         0  ...      0       0      0       1     0    0        0   \n",
       "MB-0008         0  ...      0       0      0       0     0    0        0   \n",
       "MB-0010         0  ...      1       0      0       1     0    0        0   \n",
       "...           ...  ...    ...     ...    ...     ...   ...  ...      ...   \n",
       "MTS-T2428       0  ...      1       0      0       1     0    0        0   \n",
       "MTS-T2429       0  ...      0       0      0       1     0    0        0   \n",
       "MTS-T2430       0  ...      0       0      0       1     0    0        0   \n",
       "MTS-T2431       0  ...      0       0      0       1     0    0        0   \n",
       "MTS-T2432       0  ...      1       0      0       0     0    0        0   \n",
       "\n",
       "           STK11  TBX3  TP53  \n",
       "MB-0002        0     0     1  \n",
       "MB-0005        0     0     0  \n",
       "MB-0006        0     0     0  \n",
       "MB-0008        0     0     1  \n",
       "MB-0010        0     0     1  \n",
       "...          ...   ...   ...  \n",
       "MTS-T2428      0     0     0  \n",
       "MTS-T2429      0     0     0  \n",
       "MTS-T2430      0     0     0  \n",
       "MTS-T2431      0     0     1  \n",
       "MTS-T2432      0     0     0  \n",
       "\n",
       "[2135 rows x 28 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_matrix = pd.crosstab(mut_filter['Tumor_Sample_Barcode'], mut_filter['Hugo_Symbol'])\n",
    "#mut_matrix = pd.pivot(mut_filter,index='Tumor_Sample_Barcode',values='Tumor_Sample_Barcode',columns='Hugo_Symbol').notnull()\n",
    "mut_matrix.index.name = None\n",
    "mut_matrix.columns.name = None\n",
    "mut_matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d8d9ddcc",
   "metadata": {},
   "source": [
    "Check presence of missing data, change name to columns and add 'mut' column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "392d49f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mut_AKT1</th>\n",
       "      <th>mut_ARID1A</th>\n",
       "      <th>mut_ARID1B</th>\n",
       "      <th>mut_BAP1</th>\n",
       "      <th>mut_BRCA1</th>\n",
       "      <th>mut_BRCA2</th>\n",
       "      <th>mut_BRIP1</th>\n",
       "      <th>mut_CASP8</th>\n",
       "      <th>mut_CDH1</th>\n",
       "      <th>mut_CDKN1B</th>\n",
       "      <th>...</th>\n",
       "      <th>mut_NOTCH1</th>\n",
       "      <th>mut_PBRM1</th>\n",
       "      <th>mut_PIK3CA</th>\n",
       "      <th>mut_PTEN</th>\n",
       "      <th>mut_RB1</th>\n",
       "      <th>mut_SMARCD1</th>\n",
       "      <th>mut_STK11</th>\n",
       "      <th>mut_TBX3</th>\n",
       "      <th>mut_TP53</th>\n",
       "      <th>mut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0002</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0005</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0006</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0008</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0010</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2428</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2429</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2430</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2431</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2432</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2135 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           mut_AKT1  mut_ARID1A  mut_ARID1B  mut_BAP1  mut_BRCA1  mut_BRCA2  \\\n",
       "MB-0002           0           0           0         0          0          0   \n",
       "MB-0005           0           0           0         0          0          0   \n",
       "MB-0006           0           0           0         0          0          0   \n",
       "MB-0008           0           0           0         0          0          0   \n",
       "MB-0010           0           0           0         0          0          0   \n",
       "...             ...         ...         ...       ...        ...        ...   \n",
       "MTS-T2428         0           0           0         0          0          0   \n",
       "MTS-T2429         0           0           0         0          0          0   \n",
       "MTS-T2430         0           0           0         0          0          0   \n",
       "MTS-T2431         0           0           0         0          0          0   \n",
       "MTS-T2432         0           0           0         0          0          0   \n",
       "\n",
       "           mut_BRIP1  mut_CASP8  mut_CDH1  mut_CDKN1B  ...  mut_NOTCH1  \\\n",
       "MB-0002            0          0         0           0  ...           0   \n",
       "MB-0005            0          0         0           0  ...           0   \n",
       "MB-0006            0          0         0           0  ...           0   \n",
       "MB-0008            0          0         0           0  ...           0   \n",
       "MB-0010            0          0         0           0  ...           0   \n",
       "...              ...        ...       ...         ...  ...         ...   \n",
       "MTS-T2428          0          0         0           0  ...           0   \n",
       "MTS-T2429          0          0         1           0  ...           0   \n",
       "MTS-T2430          0          0         0           0  ...           0   \n",
       "MTS-T2431          0          0         0           0  ...           0   \n",
       "MTS-T2432          0          0         0           0  ...           0   \n",
       "\n",
       "           mut_PBRM1  mut_PIK3CA  mut_PTEN  mut_RB1  mut_SMARCD1  mut_STK11  \\\n",
       "MB-0002            0           0         0        0            0          0   \n",
       "MB-0005            0           1         0        0            0          0   \n",
       "MB-0006            0           1         0        0            0          0   \n",
       "MB-0008            0           0         0        0            0          0   \n",
       "MB-0010            0           1         0        0            0          0   \n",
       "...              ...         ...       ...      ...          ...        ...   \n",
       "MTS-T2428          0           1         0        0            0          0   \n",
       "MTS-T2429          0           1         0        0            0          0   \n",
       "MTS-T2430          0           1         0        0            0          0   \n",
       "MTS-T2431          0           1         0        0            0          0   \n",
       "MTS-T2432          0           0         0        0            0          0   \n",
       "\n",
       "           mut_TBX3  mut_TP53   mut  \n",
       "MB-0002           0         1  True  \n",
       "MB-0005           0         0  True  \n",
       "MB-0006           0         0  True  \n",
       "MB-0008           0         1  True  \n",
       "MB-0010           0         1  True  \n",
       "...             ...       ...   ...  \n",
       "MTS-T2428         0         0  True  \n",
       "MTS-T2429         0         0  True  \n",
       "MTS-T2430         0         0  True  \n",
       "MTS-T2431         0         1  True  \n",
       "MTS-T2432         0         0  True  \n",
       "\n",
       "[2135 rows x 29 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# change column names\n",
    "mut_matrix.columns = 'mut_' + mut_matrix.columns\n",
    "# add 'mut' column indicating which patients have small mutations in the selected genes\n",
    "mut_matrix['mut'] = True\n",
    "mut_matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ab5a598",
   "metadata": {},
   "source": [
    "## cna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "2bc55ebc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of genes:  22542\n",
      "number of patients:  2173\n"
     ]
    }
   ],
   "source": [
    "print('number of genes: ',len(cna.Hugo_Symbol.unique()))\n",
    "print('number of patients: ',len(cna.columns)-2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e26bfaf5",
   "metadata": {},
   "source": [
    "Filtering cna table according to genes present in breast cancer genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "78700072",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes with copy numebr aberrations that are present in breast cancer gene list:  63\n",
      "Number of patients with copy number aberrations in the selected genes:  2173\n"
     ]
    }
   ],
   "source": [
    "# delete 'Entrez_Gene_Id' column\n",
    "cna_filter = cna.drop('Entrez_Gene_Id', axis=1).copy()\n",
    "# filter rows corresponding to selected genes \n",
    "cna_filter = cna_filter[cna_filter['Hugo_Symbol'].isin(gene_list)]\n",
    "print('Number of genes with copy numebr aberrations that are present in breast cancer gene list: ',len(cna_filter.Hugo_Symbol.unique()))\n",
    "print('Number of patients with copy number aberrations in the selected genes: ',len(cna_filter.columns.unique())-1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9fbee6eb",
   "metadata": {},
   "source": [
    "Create matrix with patients as rows and genes as columns and check presence of missing data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "4e9b9204",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Presence of missing data:  False\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AKT1</th>\n",
       "      <th>AR</th>\n",
       "      <th>ARID1A</th>\n",
       "      <th>ARID1B</th>\n",
       "      <th>ASPM</th>\n",
       "      <th>ATM</th>\n",
       "      <th>ATRIP</th>\n",
       "      <th>BAP1</th>\n",
       "      <th>BARD1</th>\n",
       "      <th>BRCA1</th>\n",
       "      <th>...</th>\n",
       "      <th>RAD54L</th>\n",
       "      <th>RB1</th>\n",
       "      <th>RRAS2</th>\n",
       "      <th>SALL4</th>\n",
       "      <th>SMARCD1</th>\n",
       "      <th>STK11</th>\n",
       "      <th>TBX3</th>\n",
       "      <th>TP53</th>\n",
       "      <th>XRCC2</th>\n",
       "      <th>ZMYM3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0000</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0039</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0045</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0046</th>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0048</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-6020</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-6213</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-6230</th>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7148</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7188</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2173 rows × 63 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         AKT1  AR  ARID1A  ARID1B  ASPM  ATM  ATRIP  BAP1  BARD1  BRCA1  ...  \\\n",
       "MB-0000     0   0       0       0     0    0      0     0      0      0  ...   \n",
       "MB-0039     0   0       0      -1     0   -1      0     0      0      0  ...   \n",
       "MB-0045     1   0      -1      -1     0   -1     -1    -1      0      0  ...   \n",
       "MB-0046     0  -1       0       0     2    0      0     0      0     -1  ...   \n",
       "MB-0048     0   1       0       0     1   -1      0     0      0     -1  ...   \n",
       "...       ...  ..     ...     ...   ...  ...    ...   ...    ...    ...  ...   \n",
       "MB-6020     0   0       0       0     1   -1      1     1      0     -1  ...   \n",
       "MB-6213     0   0       0       0     0    0      0     0      0      0  ...   \n",
       "MB-6230    -1   0       0      -1     1    0      0     0      0      0  ...   \n",
       "MB-7148     0   0       0       0     1    0     -1    -1      0      0  ...   \n",
       "MB-7188     0   1       0       0     0   -1      0     0      0      0  ...   \n",
       "\n",
       "         RAD54L  RB1  RRAS2  SALL4  SMARCD1  STK11  TBX3  TP53  XRCC2  ZMYM3  \n",
       "MB-0000       0    0      0      0        0      0     0     0      0      0  \n",
       "MB-0039       0   -1      0      0        0      0     0     0     -1      0  \n",
       "MB-0045      -1   -1      0      0       -1      0     0    -1      0      0  \n",
       "MB-0046       0   -1      0      1        0     -1     1    -1      0     -1  \n",
       "MB-0048       0    0      0      0        0     -1     0    -1      1      1  \n",
       "...         ...  ...    ...    ...      ...    ...   ...   ...    ...    ...  \n",
       "MB-6020       0    0      0      1        0      0     0    -1      0      0  \n",
       "MB-6213       0    0      0      0        0      0     0     0      0      0  \n",
       "MB-6230      -1    0      0      0        0      0     0     0      0      0  \n",
       "MB-7148       0    0      0      0        0      0     0     0      0      0  \n",
       "MB-7188       0    0      0      1        0      0     0     0     -1      1  \n",
       "\n",
       "[2173 rows x 63 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# transpose index and columns\n",
    "cna_filter = cna_filter.T\n",
    "cna_filter.columns = cna_filter.iloc[0]\n",
    "cna_filter = cna_filter.drop('Hugo_Symbol',axis=0)\n",
    "cna_filter.columns.name = None\n",
    "cna_filter = cna_filter.astype(int)\n",
    "# check presence of missing data\n",
    "nan = cna_filter.isnull().values.any()\n",
    "print('Presence of missing data: ',nan)\n",
    "\n",
    "cna_filter"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fce14bf5",
   "metadata": {},
   "source": [
    "Create new matrix with patients as rows and information about deletions and duplications stored in two different columns for each gene. Add 'cna' column."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "8fade573",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\simacri\\AppData\\Local\\Temp\\ipykernel_232\\2234565499.py:16: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  cna_matrix['cna'] = True\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cna_dup_AKT1</th>\n",
       "      <th>cna_del_AKT1</th>\n",
       "      <th>cna_dup_AR</th>\n",
       "      <th>cna_del_AR</th>\n",
       "      <th>cna_dup_ARID1A</th>\n",
       "      <th>cna_del_ARID1A</th>\n",
       "      <th>cna_dup_ARID1B</th>\n",
       "      <th>cna_del_ARID1B</th>\n",
       "      <th>cna_dup_ASPM</th>\n",
       "      <th>cna_del_ASPM</th>\n",
       "      <th>...</th>\n",
       "      <th>cna_del_STK11</th>\n",
       "      <th>cna_dup_TBX3</th>\n",
       "      <th>cna_del_TBX3</th>\n",
       "      <th>cna_dup_TP53</th>\n",
       "      <th>cna_del_TP53</th>\n",
       "      <th>cna_dup_XRCC2</th>\n",
       "      <th>cna_del_XRCC2</th>\n",
       "      <th>cna_dup_ZMYM3</th>\n",
       "      <th>cna_del_ZMYM3</th>\n",
       "      <th>cna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0000</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0039</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0045</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0046</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0048</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-6020</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-6213</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-6230</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7148</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7188</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2173 rows × 127 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         cna_dup_AKT1  cna_del_AKT1  cna_dup_AR  cna_del_AR  cna_dup_ARID1A  \\\n",
       "MB-0000             0             0           0           0               0   \n",
       "MB-0039             0             0           0           0               0   \n",
       "MB-0045             1             0           0           0               0   \n",
       "MB-0046             0             0           0           1               0   \n",
       "MB-0048             0             0           1           0               0   \n",
       "...               ...           ...         ...         ...             ...   \n",
       "MB-6020             0             0           0           0               0   \n",
       "MB-6213             0             0           0           0               0   \n",
       "MB-6230             0             1           0           0               0   \n",
       "MB-7148             0             0           0           0               0   \n",
       "MB-7188             0             0           1           0               0   \n",
       "\n",
       "         cna_del_ARID1A  cna_dup_ARID1B  cna_del_ARID1B  cna_dup_ASPM  \\\n",
       "MB-0000               0               0               0             0   \n",
       "MB-0039               0               0               1             0   \n",
       "MB-0045               1               0               1             0   \n",
       "MB-0046               0               0               0             1   \n",
       "MB-0048               0               0               0             1   \n",
       "...                 ...             ...             ...           ...   \n",
       "MB-6020               0               0               0             1   \n",
       "MB-6213               0               0               0             0   \n",
       "MB-6230               0               0               1             1   \n",
       "MB-7148               0               0               0             1   \n",
       "MB-7188               0               0               0             0   \n",
       "\n",
       "         cna_del_ASPM  ...  cna_del_STK11  cna_dup_TBX3  cna_del_TBX3  \\\n",
       "MB-0000             0  ...              0             0             0   \n",
       "MB-0039             0  ...              0             0             0   \n",
       "MB-0045             0  ...              0             0             0   \n",
       "MB-0046             0  ...              1             1             0   \n",
       "MB-0048             0  ...              1             0             0   \n",
       "...               ...  ...            ...           ...           ...   \n",
       "MB-6020             0  ...              0             0             0   \n",
       "MB-6213             0  ...              0             0             0   \n",
       "MB-6230             0  ...              0             0             0   \n",
       "MB-7148             0  ...              0             0             0   \n",
       "MB-7188             0  ...              0             0             0   \n",
       "\n",
       "         cna_dup_TP53  cna_del_TP53  cna_dup_XRCC2  cna_del_XRCC2  \\\n",
       "MB-0000             0             0              0              0   \n",
       "MB-0039             0             0              0              1   \n",
       "MB-0045             0             1              0              0   \n",
       "MB-0046             0             1              0              0   \n",
       "MB-0048             0             1              1              0   \n",
       "...               ...           ...            ...            ...   \n",
       "MB-6020             0             1              0              0   \n",
       "MB-6213             0             0              0              0   \n",
       "MB-6230             0             0              0              0   \n",
       "MB-7148             0             0              0              0   \n",
       "MB-7188             0             0              0              1   \n",
       "\n",
       "         cna_dup_ZMYM3  cna_del_ZMYM3   cna  \n",
       "MB-0000              0              0  True  \n",
       "MB-0039              0              0  True  \n",
       "MB-0045              0              0  True  \n",
       "MB-0046              0              1  True  \n",
       "MB-0048              1              0  True  \n",
       "...                ...            ...   ...  \n",
       "MB-6020              0              0  True  \n",
       "MB-6213              0              0  True  \n",
       "MB-6230              0              0  True  \n",
       "MB-7148              0              0  True  \n",
       "MB-7188              1              0  True  \n",
       "\n",
       "[2173 rows x 127 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cna_matrix = pd.DataFrame()\n",
    "for i in cna_filter.columns:\n",
    "    \n",
    "    label_dup = 'cna_dup_'+i\n",
    "    cna_dup = pd.DataFrame(np.where(cna_filter[i]>0,1,0))\n",
    "    cna_dup.columns = [label_dup]\n",
    "\n",
    "    label_del = 'cna_del_'+i\n",
    "    cna_del = pd.DataFrame(np.where(cna_filter[i]<0,1,0))\n",
    "    cna_del.columns = [label_del]\n",
    "    \n",
    "    cna_matrix = pd.concat([cna_matrix,cna_dup,cna_del],axis=1)\n",
    "cna_matrix.index = cna_filter.index\n",
    "\n",
    "# add 'cna' column indicating which patients have copy number variations in the selected genes\n",
    "cna_matrix['cna'] = True\n",
    "cna_matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0221c463",
   "metadata": {},
   "source": [
    "## rna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "89699959",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of genes:  20387\n",
      "number of patients:  1980\n"
     ]
    }
   ],
   "source": [
    "print('number of genes: ',len(rna.Hugo_Symbol.unique()))\n",
    "print('number of patients: ',len(rna.columns.unique())-2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d6bd77c",
   "metadata": {},
   "source": [
    "Filter rna table according to brast cancer gene list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "b75650c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of breast cancer genes with rna expression data:  63\n",
      "Number of patients with rna expression data:  1980\n"
     ]
    }
   ],
   "source": [
    "# delete 'Entrez_Gene_Id' column\n",
    "rna_filter = rna.drop('Entrez_Gene_Id', axis=1).copy()\n",
    "# filter rows corresponding to selected genes \n",
    "rna_filter = rna_filter[rna_filter['Hugo_Symbol'].isin(gene_list)]\n",
    "print('Number of breast cancer genes with rna expression data: ',len(rna_filter.Hugo_Symbol.unique()))\n",
    "print('Number of patients with rna expression data: ',len(rna_filter.columns.unique())-1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3915d484",
   "metadata": {},
   "source": [
    "Verify presence of dupicated genes with different espression values and remove them. Check presence of missing data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "7834f8ff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genes with different expression values for each patients:  ['ARID1B']\n",
      "Presence of missing data:  False\n"
     ]
    }
   ],
   "source": [
    "# verify presence of duplicated genes\n",
    "rna_filter = rna_filter.drop_duplicates()\n",
    "dupl_gene_list = []\n",
    "for i in rna_filter['Hugo_Symbol'].unique():\n",
    "    rna_single_gene = rna_filter.query(\"Hugo_Symbol=='\"+i+\"'\")\n",
    "    if (len(rna_single_gene)>1) and (len(rna_single_gene.drop_duplicates())>1):\n",
    "        dupl_gene_list.append(i)\n",
    "print('Genes with different expression values for each patients: ',dupl_gene_list)\n",
    "# delete genes \n",
    "rna_filter.index = rna_filter.Hugo_Symbol\n",
    "if len(dupl_gene_list) > 0:\n",
    "    for i in dupl_gene_list:\n",
    "        rna_filter = rna_filter.drop(i,axis=0)\n",
    "        \n",
    "# check presence of missing data\n",
    "nan = rna_filter.isnull().values.any()\n",
    "print('Presence of missing data: ',nan)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "562b1261",
   "metadata": {},
   "source": [
    "Create new matrix with patients as rows and genes as columns. Add 'rna' column."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "af88c51a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rna_NCOR1</th>\n",
       "      <th>rna_HGF</th>\n",
       "      <th>rna_FOXA1</th>\n",
       "      <th>rna_MSH6</th>\n",
       "      <th>rna_CDKN1B</th>\n",
       "      <th>rna_CDH1</th>\n",
       "      <th>rna_FADD</th>\n",
       "      <th>rna_RAD50</th>\n",
       "      <th>rna_BRCA2</th>\n",
       "      <th>rna_IRS4</th>\n",
       "      <th>...</th>\n",
       "      <th>rna_ARID1A</th>\n",
       "      <th>rna_RB1</th>\n",
       "      <th>rna_MSH2</th>\n",
       "      <th>rna_RRAS2</th>\n",
       "      <th>rna_FBLN2</th>\n",
       "      <th>rna_ATM</th>\n",
       "      <th>rna_IKZF3</th>\n",
       "      <th>rna_TBX3</th>\n",
       "      <th>rna_CHEK2</th>\n",
       "      <th>rna_EP300</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0362</th>\n",
       "      <td>1.0688</td>\n",
       "      <td>-0.615</td>\n",
       "      <td>0.6157</td>\n",
       "      <td>-1.1732</td>\n",
       "      <td>-0.9085</td>\n",
       "      <td>0.2545</td>\n",
       "      <td>2.0512</td>\n",
       "      <td>0.8194</td>\n",
       "      <td>0.656</td>\n",
       "      <td>-0.6345</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9674</td>\n",
       "      <td>0.0136</td>\n",
       "      <td>-0.7184</td>\n",
       "      <td>-0.8378</td>\n",
       "      <td>0.2111</td>\n",
       "      <td>-1.8378</td>\n",
       "      <td>-0.6141</td>\n",
       "      <td>-0.6923</td>\n",
       "      <td>-0.4167</td>\n",
       "      <td>0.6118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0346</th>\n",
       "      <td>0.7793</td>\n",
       "      <td>0.0869</td>\n",
       "      <td>0.0284</td>\n",
       "      <td>-0.8897</td>\n",
       "      <td>-1.6337</td>\n",
       "      <td>1.797</td>\n",
       "      <td>0.3542</td>\n",
       "      <td>0.4059</td>\n",
       "      <td>1.234</td>\n",
       "      <td>-0.5464</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.4781</td>\n",
       "      <td>-0.5372</td>\n",
       "      <td>-1.1661</td>\n",
       "      <td>0.0306</td>\n",
       "      <td>-1.3844</td>\n",
       "      <td>-2.8885</td>\n",
       "      <td>1.2891</td>\n",
       "      <td>-0.1955</td>\n",
       "      <td>1.8244</td>\n",
       "      <td>0.7471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0386</th>\n",
       "      <td>0.1734</td>\n",
       "      <td>2.4248</td>\n",
       "      <td>0.3589</td>\n",
       "      <td>-2.4632</td>\n",
       "      <td>-1.063</td>\n",
       "      <td>-0.3024</td>\n",
       "      <td>-0.4019</td>\n",
       "      <td>0.2079</td>\n",
       "      <td>-0.6626</td>\n",
       "      <td>1.0924</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2096</td>\n",
       "      <td>0.6206</td>\n",
       "      <td>-1.3387</td>\n",
       "      <td>-0.1813</td>\n",
       "      <td>1.1368</td>\n",
       "      <td>-1.2912</td>\n",
       "      <td>0.1893</td>\n",
       "      <td>0.2539</td>\n",
       "      <td>-0.7337</td>\n",
       "      <td>0.5146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0574</th>\n",
       "      <td>0.7201</td>\n",
       "      <td>-0.6548</td>\n",
       "      <td>0.2738</td>\n",
       "      <td>-2.1186</td>\n",
       "      <td>-1.4174</td>\n",
       "      <td>1.513</td>\n",
       "      <td>1.7116</td>\n",
       "      <td>1.3754</td>\n",
       "      <td>-1.0709</td>\n",
       "      <td>-1.9681</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.7116</td>\n",
       "      <td>1.8941</td>\n",
       "      <td>-0.7967</td>\n",
       "      <td>-0.6906</td>\n",
       "      <td>-0.2951</td>\n",
       "      <td>-1.2406</td>\n",
       "      <td>-0.7358</td>\n",
       "      <td>0.7687</td>\n",
       "      <td>0.7877</td>\n",
       "      <td>0.5379</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0185</th>\n",
       "      <td>-0.2742</td>\n",
       "      <td>1.684</td>\n",
       "      <td>0.4365</td>\n",
       "      <td>0.1175</td>\n",
       "      <td>-0.7677</td>\n",
       "      <td>0.4657</td>\n",
       "      <td>0.4333</td>\n",
       "      <td>-1.1464</td>\n",
       "      <td>-0.0708</td>\n",
       "      <td>-0.218</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0249</td>\n",
       "      <td>0.3192</td>\n",
       "      <td>-0.546</td>\n",
       "      <td>1.3484</td>\n",
       "      <td>-1.4457</td>\n",
       "      <td>-0.7969</td>\n",
       "      <td>1.1604</td>\n",
       "      <td>0.6501</td>\n",
       "      <td>4.2078</td>\n",
       "      <td>0.7796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-5453</th>\n",
       "      <td>0.8373</td>\n",
       "      <td>-2.453</td>\n",
       "      <td>-0.3096</td>\n",
       "      <td>0.5525</td>\n",
       "      <td>1.4565</td>\n",
       "      <td>-0.1619</td>\n",
       "      <td>0.2785</td>\n",
       "      <td>0.8496</td>\n",
       "      <td>-0.5676</td>\n",
       "      <td>-0.34</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8886</td>\n",
       "      <td>-0.0685</td>\n",
       "      <td>0.5833</td>\n",
       "      <td>0.3933</td>\n",
       "      <td>-1.3792</td>\n",
       "      <td>1.6544</td>\n",
       "      <td>-1.188</td>\n",
       "      <td>0.6562</td>\n",
       "      <td>-0.5058</td>\n",
       "      <td>-0.5387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-5471</th>\n",
       "      <td>-0.406</td>\n",
       "      <td>-1.2909</td>\n",
       "      <td>0.3128</td>\n",
       "      <td>0.5762</td>\n",
       "      <td>0.6839</td>\n",
       "      <td>0.5722</td>\n",
       "      <td>-0.7621</td>\n",
       "      <td>-0.4317</td>\n",
       "      <td>0.7238</td>\n",
       "      <td>0.44</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0488</td>\n",
       "      <td>0.1549</td>\n",
       "      <td>-0.3421</td>\n",
       "      <td>-0.544</td>\n",
       "      <td>0.5705</td>\n",
       "      <td>0.7298</td>\n",
       "      <td>-0.5398</td>\n",
       "      <td>-0.0435</td>\n",
       "      <td>2.5491</td>\n",
       "      <td>1.2823</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-5127</th>\n",
       "      <td>1.7295</td>\n",
       "      <td>0.9662</td>\n",
       "      <td>0.5625</td>\n",
       "      <td>0.8419</td>\n",
       "      <td>2.1601</td>\n",
       "      <td>0.7309</td>\n",
       "      <td>1.6882</td>\n",
       "      <td>-0.0815</td>\n",
       "      <td>0.3955</td>\n",
       "      <td>-0.5292</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.5412</td>\n",
       "      <td>0.734</td>\n",
       "      <td>0.9264</td>\n",
       "      <td>-1.9107</td>\n",
       "      <td>-1.6665</td>\n",
       "      <td>0.4067</td>\n",
       "      <td>-0.1244</td>\n",
       "      <td>-0.8301</td>\n",
       "      <td>0.2351</td>\n",
       "      <td>-0.9663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-4313</th>\n",
       "      <td>-2.9275</td>\n",
       "      <td>-0.5509</td>\n",
       "      <td>-0.7498</td>\n",
       "      <td>-0.6392</td>\n",
       "      <td>-0.8716</td>\n",
       "      <td>-0.8985</td>\n",
       "      <td>-1.9611</td>\n",
       "      <td>-3.0446</td>\n",
       "      <td>0.4947</td>\n",
       "      <td>3.9558</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.0725</td>\n",
       "      <td>-1.0741</td>\n",
       "      <td>-1.3687</td>\n",
       "      <td>-1.3555</td>\n",
       "      <td>-0.2023</td>\n",
       "      <td>-1.1483</td>\n",
       "      <td>0.3596</td>\n",
       "      <td>0.0385</td>\n",
       "      <td>-2.3226</td>\n",
       "      <td>-4.377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-4823</th>\n",
       "      <td>-0.3772</td>\n",
       "      <td>-1.2197</td>\n",
       "      <td>0.5127</td>\n",
       "      <td>0.5987</td>\n",
       "      <td>0.1724</td>\n",
       "      <td>0.0792</td>\n",
       "      <td>0.1523</td>\n",
       "      <td>0.4528</td>\n",
       "      <td>-0.8973</td>\n",
       "      <td>2.0409</td>\n",
       "      <td>...</td>\n",
       "      <td>0.7928</td>\n",
       "      <td>0.3937</td>\n",
       "      <td>0.8453</td>\n",
       "      <td>0.8648</td>\n",
       "      <td>-1.2183</td>\n",
       "      <td>0.838</td>\n",
       "      <td>1.3358</td>\n",
       "      <td>-0.7074</td>\n",
       "      <td>-0.0399</td>\n",
       "      <td>-0.5436</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1980 rows × 62 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        rna_NCOR1 rna_HGF rna_FOXA1 rna_MSH6 rna_CDKN1B rna_CDH1 rna_FADD  \\\n",
       "MB-0362    1.0688  -0.615    0.6157  -1.1732    -0.9085   0.2545   2.0512   \n",
       "MB-0346    0.7793  0.0869    0.0284  -0.8897    -1.6337    1.797   0.3542   \n",
       "MB-0386    0.1734  2.4248    0.3589  -2.4632     -1.063  -0.3024  -0.4019   \n",
       "MB-0574    0.7201 -0.6548    0.2738  -2.1186    -1.4174    1.513   1.7116   \n",
       "MB-0185   -0.2742   1.684    0.4365   0.1175    -0.7677   0.4657   0.4333   \n",
       "...           ...     ...       ...      ...        ...      ...      ...   \n",
       "MB-5453    0.8373  -2.453   -0.3096   0.5525     1.4565  -0.1619   0.2785   \n",
       "MB-5471    -0.406 -1.2909    0.3128   0.5762     0.6839   0.5722  -0.7621   \n",
       "MB-5127    1.7295  0.9662    0.5625   0.8419     2.1601   0.7309   1.6882   \n",
       "MB-4313   -2.9275 -0.5509   -0.7498  -0.6392    -0.8716  -0.8985  -1.9611   \n",
       "MB-4823   -0.3772 -1.2197    0.5127   0.5987     0.1724   0.0792   0.1523   \n",
       "\n",
       "        rna_RAD50 rna_BRCA2 rna_IRS4  ... rna_ARID1A rna_RB1 rna_MSH2  \\\n",
       "MB-0362    0.8194     0.656  -0.6345  ...     0.9674  0.0136  -0.7184   \n",
       "MB-0346    0.4059     1.234  -0.5464  ...    -0.4781 -0.5372  -1.1661   \n",
       "MB-0386    0.2079   -0.6626   1.0924  ...     0.2096  0.6206  -1.3387   \n",
       "MB-0574    1.3754   -1.0709  -1.9681  ...    -0.7116  1.8941  -0.7967   \n",
       "MB-0185   -1.1464   -0.0708   -0.218  ...     0.0249  0.3192   -0.546   \n",
       "...           ...       ...      ...  ...        ...     ...      ...   \n",
       "MB-5453    0.8496   -0.5676    -0.34  ...     0.8886 -0.0685   0.5833   \n",
       "MB-5471   -0.4317    0.7238     0.44  ...     0.0488  0.1549  -0.3421   \n",
       "MB-5127   -0.0815    0.3955  -0.5292  ...    -0.5412   0.734   0.9264   \n",
       "MB-4313   -3.0446    0.4947   3.9558  ...    -5.0725 -1.0741  -1.3687   \n",
       "MB-4823    0.4528   -0.8973   2.0409  ...     0.7928  0.3937   0.8453   \n",
       "\n",
       "        rna_RRAS2 rna_FBLN2 rna_ATM rna_IKZF3 rna_TBX3 rna_CHEK2 rna_EP300  \n",
       "MB-0362   -0.8378    0.2111 -1.8378   -0.6141  -0.6923   -0.4167    0.6118  \n",
       "MB-0346    0.0306   -1.3844 -2.8885    1.2891  -0.1955    1.8244    0.7471  \n",
       "MB-0386   -0.1813    1.1368 -1.2912    0.1893   0.2539   -0.7337    0.5146  \n",
       "MB-0574   -0.6906   -0.2951 -1.2406   -0.7358   0.7687    0.7877    0.5379  \n",
       "MB-0185    1.3484   -1.4457 -0.7969    1.1604   0.6501    4.2078    0.7796  \n",
       "...           ...       ...     ...       ...      ...       ...       ...  \n",
       "MB-5453    0.3933   -1.3792  1.6544    -1.188   0.6562   -0.5058   -0.5387  \n",
       "MB-5471    -0.544    0.5705  0.7298   -0.5398  -0.0435    2.5491    1.2823  \n",
       "MB-5127   -1.9107   -1.6665  0.4067   -0.1244  -0.8301    0.2351   -0.9663  \n",
       "MB-4313   -1.3555   -0.2023 -1.1483    0.3596   0.0385   -2.3226    -4.377  \n",
       "MB-4823    0.8648   -1.2183   0.838    1.3358  -0.7074   -0.0399   -0.5436  \n",
       "\n",
       "[1980 rows x 62 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna_matrix = rna_filter.T.copy()\n",
    "rna_matrix.columns = rna_matrix.iloc[0]\n",
    "rna_matrix.columns.name = None\n",
    "rna_matrix = rna_matrix.drop('Hugo_Symbol')\n",
    "rna_matrix.columns = 'rna_' + rna_matrix.columns\n",
    "rna_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "e98f68ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rna_NCOR1</th>\n",
       "      <th>rna_HGF</th>\n",
       "      <th>rna_FOXA1</th>\n",
       "      <th>rna_MSH6</th>\n",
       "      <th>rna_CDKN1B</th>\n",
       "      <th>rna_CDH1</th>\n",
       "      <th>rna_FADD</th>\n",
       "      <th>rna_RAD50</th>\n",
       "      <th>rna_BRCA2</th>\n",
       "      <th>rna_IRS4</th>\n",
       "      <th>...</th>\n",
       "      <th>rna_RB1</th>\n",
       "      <th>rna_MSH2</th>\n",
       "      <th>rna_RRAS2</th>\n",
       "      <th>rna_FBLN2</th>\n",
       "      <th>rna_ATM</th>\n",
       "      <th>rna_IKZF3</th>\n",
       "      <th>rna_TBX3</th>\n",
       "      <th>rna_CHEK2</th>\n",
       "      <th>rna_EP300</th>\n",
       "      <th>rna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0362</th>\n",
       "      <td>0.493942</td>\n",
       "      <td>0.393584</td>\n",
       "      <td>0.466936</td>\n",
       "      <td>0.360315</td>\n",
       "      <td>0.376091</td>\n",
       "      <td>0.445408</td>\n",
       "      <td>0.552494</td>\n",
       "      <td>0.479077</td>\n",
       "      <td>0.469338</td>\n",
       "      <td>0.392422</td>\n",
       "      <td>...</td>\n",
       "      <td>0.43105</td>\n",
       "      <td>0.387422</td>\n",
       "      <td>0.380305</td>\n",
       "      <td>0.442821</td>\n",
       "      <td>0.320704</td>\n",
       "      <td>0.393638</td>\n",
       "      <td>0.388977</td>\n",
       "      <td>0.405403</td>\n",
       "      <td>0.466704</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0346</th>\n",
       "      <td>0.476687</td>\n",
       "      <td>0.435419</td>\n",
       "      <td>0.431932</td>\n",
       "      <td>0.377212</td>\n",
       "      <td>0.332868</td>\n",
       "      <td>0.537343</td>\n",
       "      <td>0.45135</td>\n",
       "      <td>0.454432</td>\n",
       "      <td>0.503788</td>\n",
       "      <td>0.397673</td>\n",
       "      <td>...</td>\n",
       "      <td>0.398221</td>\n",
       "      <td>0.360738</td>\n",
       "      <td>0.432063</td>\n",
       "      <td>0.347727</td>\n",
       "      <td>0.25808</td>\n",
       "      <td>0.507072</td>\n",
       "      <td>0.418587</td>\n",
       "      <td>0.538976</td>\n",
       "      <td>0.474768</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0386</th>\n",
       "      <td>0.440574</td>\n",
       "      <td>0.574761</td>\n",
       "      <td>0.45163</td>\n",
       "      <td>0.283429</td>\n",
       "      <td>0.366883</td>\n",
       "      <td>0.412216</td>\n",
       "      <td>0.406286</td>\n",
       "      <td>0.442631</td>\n",
       "      <td>0.390747</td>\n",
       "      <td>0.495348</td>\n",
       "      <td>...</td>\n",
       "      <td>0.467228</td>\n",
       "      <td>0.350451</td>\n",
       "      <td>0.419434</td>\n",
       "      <td>0.497994</td>\n",
       "      <td>0.353282</td>\n",
       "      <td>0.441522</td>\n",
       "      <td>0.445372</td>\n",
       "      <td>0.38651</td>\n",
       "      <td>0.46091</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0574</th>\n",
       "      <td>0.473158</td>\n",
       "      <td>0.391212</td>\n",
       "      <td>0.446558</td>\n",
       "      <td>0.303968</td>\n",
       "      <td>0.34576</td>\n",
       "      <td>0.520416</td>\n",
       "      <td>0.532253</td>\n",
       "      <td>0.512215</td>\n",
       "      <td>0.366412</td>\n",
       "      <td>0.312938</td>\n",
       "      <td>...</td>\n",
       "      <td>0.543131</td>\n",
       "      <td>0.382755</td>\n",
       "      <td>0.389079</td>\n",
       "      <td>0.412651</td>\n",
       "      <td>0.356298</td>\n",
       "      <td>0.386385</td>\n",
       "      <td>0.476055</td>\n",
       "      <td>0.477188</td>\n",
       "      <td>0.462299</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0185</th>\n",
       "      <td>0.413897</td>\n",
       "      <td>0.530608</td>\n",
       "      <td>0.456255</td>\n",
       "      <td>0.437243</td>\n",
       "      <td>0.384483</td>\n",
       "      <td>0.457996</td>\n",
       "      <td>0.456065</td>\n",
       "      <td>0.361912</td>\n",
       "      <td>0.42602</td>\n",
       "      <td>0.417246</td>\n",
       "      <td>...</td>\n",
       "      <td>0.449264</td>\n",
       "      <td>0.397697</td>\n",
       "      <td>0.510606</td>\n",
       "      <td>0.344074</td>\n",
       "      <td>0.382743</td>\n",
       "      <td>0.499401</td>\n",
       "      <td>0.468986</td>\n",
       "      <td>0.681031</td>\n",
       "      <td>0.476705</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-5453</th>\n",
       "      <td>0.480144</td>\n",
       "      <td>0.284037</td>\n",
       "      <td>0.411787</td>\n",
       "      <td>0.463169</td>\n",
       "      <td>0.517049</td>\n",
       "      <td>0.42059</td>\n",
       "      <td>0.446838</td>\n",
       "      <td>0.480877</td>\n",
       "      <td>0.39641</td>\n",
       "      <td>0.409975</td>\n",
       "      <td>...</td>\n",
       "      <td>0.426157</td>\n",
       "      <td>0.465005</td>\n",
       "      <td>0.453681</td>\n",
       "      <td>0.348037</td>\n",
       "      <td>0.528844</td>\n",
       "      <td>0.359433</td>\n",
       "      <td>0.46935</td>\n",
       "      <td>0.400093</td>\n",
       "      <td>0.398132</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-5471</th>\n",
       "      <td>0.406041</td>\n",
       "      <td>0.3533</td>\n",
       "      <td>0.448883</td>\n",
       "      <td>0.464582</td>\n",
       "      <td>0.471001</td>\n",
       "      <td>0.464343</td>\n",
       "      <td>0.384817</td>\n",
       "      <td>0.404509</td>\n",
       "      <td>0.473379</td>\n",
       "      <td>0.456464</td>\n",
       "      <td>...</td>\n",
       "      <td>0.439472</td>\n",
       "      <td>0.40985</td>\n",
       "      <td>0.397816</td>\n",
       "      <td>0.464242</td>\n",
       "      <td>0.473737</td>\n",
       "      <td>0.398067</td>\n",
       "      <td>0.427647</td>\n",
       "      <td>0.58217</td>\n",
       "      <td>0.506666</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-5127</th>\n",
       "      <td>0.53332</td>\n",
       "      <td>0.487826</td>\n",
       "      <td>0.463765</td>\n",
       "      <td>0.480418</td>\n",
       "      <td>0.558985</td>\n",
       "      <td>0.473802</td>\n",
       "      <td>0.530859</td>\n",
       "      <td>0.425382</td>\n",
       "      <td>0.453812</td>\n",
       "      <td>0.398698</td>\n",
       "      <td>...</td>\n",
       "      <td>0.473987</td>\n",
       "      <td>0.485454</td>\n",
       "      <td>0.316359</td>\n",
       "      <td>0.330914</td>\n",
       "      <td>0.454479</td>\n",
       "      <td>0.422825</td>\n",
       "      <td>0.380764</td>\n",
       "      <td>0.444252</td>\n",
       "      <td>0.372646</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-4313</th>\n",
       "      <td>0.255756</td>\n",
       "      <td>0.397405</td>\n",
       "      <td>0.38555</td>\n",
       "      <td>0.392142</td>\n",
       "      <td>0.378291</td>\n",
       "      <td>0.376687</td>\n",
       "      <td>0.313355</td>\n",
       "      <td>0.248777</td>\n",
       "      <td>0.459724</td>\n",
       "      <td>0.666011</td>\n",
       "      <td>...</td>\n",
       "      <td>0.366221</td>\n",
       "      <td>0.348663</td>\n",
       "      <td>0.34945</td>\n",
       "      <td>0.418182</td>\n",
       "      <td>0.361799</td>\n",
       "      <td>0.451672</td>\n",
       "      <td>0.432534</td>\n",
       "      <td>0.291809</td>\n",
       "      <td>0.169364</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-4823</th>\n",
       "      <td>0.407758</td>\n",
       "      <td>0.357543</td>\n",
       "      <td>0.460797</td>\n",
       "      <td>0.465923</td>\n",
       "      <td>0.440515</td>\n",
       "      <td>0.43496</td>\n",
       "      <td>0.439317</td>\n",
       "      <td>0.457227</td>\n",
       "      <td>0.376759</td>\n",
       "      <td>0.55188</td>\n",
       "      <td>...</td>\n",
       "      <td>0.453705</td>\n",
       "      <td>0.480621</td>\n",
       "      <td>0.481783</td>\n",
       "      <td>0.357627</td>\n",
       "      <td>0.480185</td>\n",
       "      <td>0.509855</td>\n",
       "      <td>0.388077</td>\n",
       "      <td>0.427861</td>\n",
       "      <td>0.39784</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1980 rows × 63 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        rna_NCOR1   rna_HGF rna_FOXA1  rna_MSH6 rna_CDKN1B  rna_CDH1  \\\n",
       "MB-0362  0.493942  0.393584  0.466936  0.360315   0.376091  0.445408   \n",
       "MB-0346  0.476687  0.435419  0.431932  0.377212   0.332868  0.537343   \n",
       "MB-0386  0.440574  0.574761   0.45163  0.283429   0.366883  0.412216   \n",
       "MB-0574  0.473158  0.391212  0.446558  0.303968    0.34576  0.520416   \n",
       "MB-0185  0.413897  0.530608  0.456255  0.437243   0.384483  0.457996   \n",
       "...           ...       ...       ...       ...        ...       ...   \n",
       "MB-5453  0.480144  0.284037  0.411787  0.463169   0.517049   0.42059   \n",
       "MB-5471  0.406041    0.3533  0.448883  0.464582   0.471001  0.464343   \n",
       "MB-5127   0.53332  0.487826  0.463765  0.480418   0.558985  0.473802   \n",
       "MB-4313  0.255756  0.397405   0.38555  0.392142   0.378291  0.376687   \n",
       "MB-4823  0.407758  0.357543  0.460797  0.465923   0.440515   0.43496   \n",
       "\n",
       "         rna_FADD rna_RAD50 rna_BRCA2  rna_IRS4  ...   rna_RB1  rna_MSH2  \\\n",
       "MB-0362  0.552494  0.479077  0.469338  0.392422  ...   0.43105  0.387422   \n",
       "MB-0346   0.45135  0.454432  0.503788  0.397673  ...  0.398221  0.360738   \n",
       "MB-0386  0.406286  0.442631  0.390747  0.495348  ...  0.467228  0.350451   \n",
       "MB-0574  0.532253  0.512215  0.366412  0.312938  ...  0.543131  0.382755   \n",
       "MB-0185  0.456065  0.361912   0.42602  0.417246  ...  0.449264  0.397697   \n",
       "...           ...       ...       ...       ...  ...       ...       ...   \n",
       "MB-5453  0.446838  0.480877   0.39641  0.409975  ...  0.426157  0.465005   \n",
       "MB-5471  0.384817  0.404509  0.473379  0.456464  ...  0.439472   0.40985   \n",
       "MB-5127  0.530859  0.425382  0.453812  0.398698  ...  0.473987  0.485454   \n",
       "MB-4313  0.313355  0.248777  0.459724  0.666011  ...  0.366221  0.348663   \n",
       "MB-4823  0.439317  0.457227  0.376759   0.55188  ...  0.453705  0.480621   \n",
       "\n",
       "        rna_RRAS2 rna_FBLN2   rna_ATM rna_IKZF3  rna_TBX3 rna_CHEK2 rna_EP300  \\\n",
       "MB-0362  0.380305  0.442821  0.320704  0.393638  0.388977  0.405403  0.466704   \n",
       "MB-0346  0.432063  0.347727   0.25808  0.507072  0.418587  0.538976  0.474768   \n",
       "MB-0386  0.419434  0.497994  0.353282  0.441522  0.445372   0.38651   0.46091   \n",
       "MB-0574  0.389079  0.412651  0.356298  0.386385  0.476055  0.477188  0.462299   \n",
       "MB-0185  0.510606  0.344074  0.382743  0.499401  0.468986  0.681031  0.476705   \n",
       "...           ...       ...       ...       ...       ...       ...       ...   \n",
       "MB-5453  0.453681  0.348037  0.528844  0.359433   0.46935  0.400093  0.398132   \n",
       "MB-5471  0.397816  0.464242  0.473737  0.398067  0.427647   0.58217  0.506666   \n",
       "MB-5127  0.316359  0.330914  0.454479  0.422825  0.380764  0.444252  0.372646   \n",
       "MB-4313   0.34945  0.418182  0.361799  0.451672  0.432534  0.291809  0.169364   \n",
       "MB-4823  0.481783  0.357627  0.480185  0.509855  0.388077  0.427861   0.39784   \n",
       "\n",
       "          rna  \n",
       "MB-0362  True  \n",
       "MB-0346  True  \n",
       "MB-0386  True  \n",
       "MB-0574  True  \n",
       "MB-0185  True  \n",
       "...       ...  \n",
       "MB-5453  True  \n",
       "MB-5471  True  \n",
       "MB-5127  True  \n",
       "MB-4313  True  \n",
       "MB-4823  True  \n",
       "\n",
       "[1980 rows x 63 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# scaling to [0-1]\n",
    "rna_matrix = (rna_matrix-rna_matrix.min().min())/(rna_matrix.max().max()-rna_matrix.min().min())\n",
    "# add 'rna' column indicating which patients have rna expression data\n",
    "rna_matrix['rna'] = True\n",
    "rna_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "1b4c1dbb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna_matrix.max().max()\n",
    "#rna_matrix.min().min()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b9f69fe2",
   "metadata": {},
   "source": [
    "## meth "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "ee9c3819",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of genes:  13184\n",
      "number of patients:  1417\n"
     ]
    }
   ],
   "source": [
    "print('number of genes: ',len(meth.Hugo_Symbol.unique()))\n",
    "print('number of patients: ',len(meth.columns.unique())-2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a72442ed",
   "metadata": {},
   "source": [
    "Filter methylation data using the breast cancer gene list."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "c5904805",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of breast cancer genes with methylation data:  54\n",
      "Number of patients with methylation data:  1418\n"
     ]
    }
   ],
   "source": [
    "meth_filter = meth[meth['Hugo_Symbol'].isin(gene_list)].copy()\n",
    "meth_filter = meth_filter.T.copy()\n",
    "meth_filter.columns = meth_filter.iloc[0]\n",
    "meth_filter.columns.name = None\n",
    "meth_filter = meth_filter.drop('Hugo_Symbol')\n",
    "meth_filter.columns = 'meth_' + meth_filter.columns\n",
    "print('Number of breast cancer genes with methylation data: ',len(meth_filter.columns.unique()))\n",
    "print('Number of patients with methylation data: ',len(meth_filter.index.unique()))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0f7cc6a",
   "metadata": {},
   "source": [
    "Check the presence of missing data and perform imputation by substituting missing data of a selected gene with the mean value over patients of the other data corresponding to the same gene. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "6a529b3e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Presence of missing data:  True\n",
      "Percentage of nan over the total amount of data:  4.347542182521026\n"
     ]
    }
   ],
   "source": [
    "# check presence of missing data\n",
    "nan = meth_filter.isnull().values.any()\n",
    "print('Presence of missing data: ',nan)\n",
    "nan_tot = meth_filter.isnull().sum().sum()\n",
    "perc = nan_tot*100/(len(meth_filter.columns)*len(meth_filter))\n",
    "print('Percentage of nan over the total amount of data: ',perc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "c6d2a932",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Presence of missing data after imputation:  False\n"
     ]
    }
   ],
   "source": [
    "# imputation with mean value over patients\n",
    "meth_matrix = meth_filter.copy()\n",
    "for i in meth_matrix.columns:\n",
    "    if meth_matrix[i].isnull().values.any() == False:\n",
    "        continue\n",
    "    mean = meth_matrix[i][meth_matrix[i].notna()].mean()\n",
    "    meth_matrix[i][meth_matrix[i].isna()] = mean\n",
    "nan = meth_matrix.isnull().values.any()\n",
    "print('Presence of missing data after imputation: ',nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "9227d929",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>meth_AKT1</th>\n",
       "      <th>meth_AR</th>\n",
       "      <th>meth_ARID1A</th>\n",
       "      <th>meth_ARID1B</th>\n",
       "      <th>meth_ASPM</th>\n",
       "      <th>meth_ATM</th>\n",
       "      <th>meth_ATRIP</th>\n",
       "      <th>meth_BAP1</th>\n",
       "      <th>meth_BARD1</th>\n",
       "      <th>meth_BRCA1</th>\n",
       "      <th>...</th>\n",
       "      <th>meth_RB1</th>\n",
       "      <th>meth_RRAS2</th>\n",
       "      <th>meth_SALL4</th>\n",
       "      <th>meth_SMARCD1</th>\n",
       "      <th>meth_STK11</th>\n",
       "      <th>meth_TBX3</th>\n",
       "      <th>meth_TP53</th>\n",
       "      <th>meth_XRCC2</th>\n",
       "      <th>meth_ZMYM3</th>\n",
       "      <th>meth</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0006</th>\n",
       "      <td>0.012346</td>\n",
       "      <td>0.345133</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.014706</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002094</td>\n",
       "      <td>0.007246</td>\n",
       "      <td>0.374549</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.079208</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00365</td>\n",
       "      <td>0.348101</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0028</th>\n",
       "      <td>0.042254</td>\n",
       "      <td>0.095238</td>\n",
       "      <td>0.008048</td>\n",
       "      <td>0.002353</td>\n",
       "      <td>0.011111</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005814</td>\n",
       "      <td>0.018692</td>\n",
       "      <td>0.002066</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.005792</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.737374</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.058824</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.007143</td>\n",
       "      <td>0.145455</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0035</th>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001468</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.003279</td>\n",
       "      <td>0.015933</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.047583</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.875949</td>\n",
       "      <td>0.016287</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.507246</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0046</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.214286</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00108</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.379421</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.004902</td>\n",
       "      <td>0.239583</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0050</th>\n",
       "      <td>0.032615</td>\n",
       "      <td>0.263158</td>\n",
       "      <td>0.00141</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.004098</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002632</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.047583</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001374</td>\n",
       "      <td>0.005063</td>\n",
       "      <td>0.572327</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.098446</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7289</th>\n",
       "      <td>0.025</td>\n",
       "      <td>0.314815</td>\n",
       "      <td>0.001196</td>\n",
       "      <td>0.003783</td>\n",
       "      <td>0.007067</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001653</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002601</td>\n",
       "      <td>0.063291</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.695364</td>\n",
       "      <td>0.005848</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00607</td>\n",
       "      <td>0.242915</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7291</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.30303</td>\n",
       "      <td>0.001185</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001435</td>\n",
       "      <td>0.015933</td>\n",
       "      <td>0.002861</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001116</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.579268</td>\n",
       "      <td>0.002268</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.028571</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001408</td>\n",
       "      <td>0.201754</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7292</th>\n",
       "      <td>0.032615</td>\n",
       "      <td>0.291393</td>\n",
       "      <td>0.006309</td>\n",
       "      <td>0.007782</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005076</td>\n",
       "      <td>0.015933</td>\n",
       "      <td>0.003623</td>\n",
       "      <td>0.02439</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02509</td>\n",
       "      <td>0.568627</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00641</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.004141</td>\n",
       "      <td>0.00995</td>\n",
       "      <td>0.21374</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7293</th>\n",
       "      <td>0.017857</td>\n",
       "      <td>0.291393</td>\n",
       "      <td>0.001594</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009901</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001592</td>\n",
       "      <td>0.047583</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.468599</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.002878</td>\n",
       "      <td>0.39196</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-7296</th>\n",
       "      <td>0.032615</td>\n",
       "      <td>0.291393</td>\n",
       "      <td>0.007895</td>\n",
       "      <td>0.007519</td>\n",
       "      <td>0.029703</td>\n",
       "      <td>0.030303</td>\n",
       "      <td>0.005421</td>\n",
       "      <td>0.015933</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.047583</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.830769</td>\n",
       "      <td>0.015075</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.083333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.255474</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1418 rows × 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        meth_AKT1   meth_AR meth_ARID1A meth_ARID1B meth_ASPM  meth_ATM  \\\n",
       "MB-0006  0.012346  0.345133         0.0         0.0       0.0       0.0   \n",
       "MB-0028  0.042254  0.095238    0.008048    0.002353  0.011111       0.0   \n",
       "MB-0035  0.005587      0.15         0.0    0.001468       0.0       0.0   \n",
       "MB-0046       0.0  0.214286         0.0     0.00108       0.0       0.0   \n",
       "MB-0050  0.032615  0.263158     0.00141         0.0  0.004098       0.0   \n",
       "...           ...       ...         ...         ...       ...       ...   \n",
       "MB-7289     0.025  0.314815    0.001196    0.003783  0.007067       0.0   \n",
       "MB-7291       0.0   0.30303    0.001185         0.0       0.0       0.0   \n",
       "MB-7292  0.032615  0.291393    0.006309    0.007782       0.0       0.0   \n",
       "MB-7293  0.017857  0.291393    0.001594         0.0       0.0  0.009901   \n",
       "MB-7296  0.032615  0.291393    0.007895    0.007519  0.029703  0.030303   \n",
       "\n",
       "        meth_ATRIP meth_BAP1 meth_BARD1 meth_BRCA1  ...  meth_RB1 meth_RRAS2  \\\n",
       "MB-0006        0.0       0.0        0.0   0.014706  ...  0.002094   0.007246   \n",
       "MB-0028   0.005814  0.018692   0.002066        0.0  ...  0.005792        0.0   \n",
       "MB-0035   0.003279  0.015933        0.0   0.047583  ...       0.0        0.0   \n",
       "MB-0046        0.0       0.0        0.0        0.0  ...       0.0        0.0   \n",
       "MB-0050   0.002632       0.0        0.0   0.047583  ...  0.001374   0.005063   \n",
       "...            ...       ...        ...        ...  ...       ...        ...   \n",
       "MB-7289   0.001653       0.0   0.002601   0.063291  ...       0.0        0.0   \n",
       "MB-7291   0.001435  0.015933   0.002861        0.0  ...  0.001116        0.0   \n",
       "MB-7292   0.005076  0.015933   0.003623    0.02439  ...       0.0    0.02509   \n",
       "MB-7293        0.0       0.0   0.001592   0.047583  ...       0.0        0.0   \n",
       "MB-7296   0.005421  0.015933        0.0   0.047583  ...       0.0        0.0   \n",
       "\n",
       "        meth_SALL4 meth_SMARCD1 meth_STK11 meth_TBX3 meth_TP53 meth_XRCC2  \\\n",
       "MB-0006   0.374549          0.0        0.0  0.079208       0.0    0.00365   \n",
       "MB-0028   0.737374          0.0        0.0  0.058824       0.0   0.007143   \n",
       "MB-0035   0.875949     0.016287        0.0       0.0       0.0        0.0   \n",
       "MB-0046   0.379421          0.0        0.0       0.0       0.0   0.004902   \n",
       "MB-0050   0.572327          0.0        0.0       0.0       0.0        0.0   \n",
       "...            ...          ...        ...       ...       ...        ...   \n",
       "MB-7289   0.695364     0.005848        0.0       0.0       0.0    0.00607   \n",
       "MB-7291   0.579268     0.002268        0.0  0.028571       0.0   0.001408   \n",
       "MB-7292   0.568627          0.0    0.00641       0.0  0.004141    0.00995   \n",
       "MB-7293   0.468599          0.0        0.0       0.0       0.0   0.002878   \n",
       "MB-7296   0.830769     0.015075        0.0  0.083333       0.0        0.0   \n",
       "\n",
       "        meth_ZMYM3  meth  \n",
       "MB-0006   0.348101  True  \n",
       "MB-0028   0.145455  True  \n",
       "MB-0035   0.507246  True  \n",
       "MB-0046   0.239583  True  \n",
       "MB-0050   0.098446  True  \n",
       "...            ...   ...  \n",
       "MB-7289   0.242915  True  \n",
       "MB-7291   0.201754  True  \n",
       "MB-7292    0.21374  True  \n",
       "MB-7293    0.39196  True  \n",
       "MB-7296   0.255474  True  \n",
       "\n",
       "[1418 rows x 55 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# scaling to [0-1]\n",
    "meth_matrix = (meth_matrix-meth_matrix.min().min())/(meth_matrix.max().max()-meth_matrix.min().min())\n",
    "# add 'meth' column indicating which patients have methylation data\n",
    "meth_matrix['meth'] = True\n",
    "meth_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "996115a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meth_matrix.max().max()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b70138c7",
   "metadata": {},
   "source": [
    "## Merge all genomic data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "e19d69e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mut_AKT1</th>\n",
       "      <th>mut_ARID1A</th>\n",
       "      <th>mut_ARID1B</th>\n",
       "      <th>mut_BAP1</th>\n",
       "      <th>mut_BRCA1</th>\n",
       "      <th>mut_BRCA2</th>\n",
       "      <th>mut_BRIP1</th>\n",
       "      <th>mut_CASP8</th>\n",
       "      <th>mut_CDH1</th>\n",
       "      <th>mut_CDKN1B</th>\n",
       "      <th>...</th>\n",
       "      <th>meth_SMARCD1</th>\n",
       "      <th>meth_STK11</th>\n",
       "      <th>meth_TBX3</th>\n",
       "      <th>meth_TP53</th>\n",
       "      <th>meth_XRCC2</th>\n",
       "      <th>meth_ZMYM3</th>\n",
       "      <th>mut</th>\n",
       "      <th>cna</th>\n",
       "      <th>rna</th>\n",
       "      <th>meth</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0000</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0002</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0005</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0006</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.079208</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00365</td>\n",
       "      <td>0.348101</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0008</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2428</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2429</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2430</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2431</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2432</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2454 rows × 274 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           mut_AKT1  mut_ARID1A  mut_ARID1B  mut_BAP1  mut_BRCA1  mut_BRCA2  \\\n",
       "MB-0000         NaN         NaN         NaN       NaN        NaN        NaN   \n",
       "MB-0002         0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MB-0005         0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MB-0006         0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MB-0008         0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "...             ...         ...         ...       ...        ...        ...   \n",
       "MTS-T2428       0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MTS-T2429       0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MTS-T2430       0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MTS-T2431       0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MTS-T2432       0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "\n",
       "           mut_BRIP1  mut_CASP8  mut_CDH1  mut_CDKN1B  ...  meth_SMARCD1  \\\n",
       "MB-0000          NaN        NaN       NaN         NaN  ...           NaN   \n",
       "MB-0002          0.0        0.0       0.0         0.0  ...           NaN   \n",
       "MB-0005          0.0        0.0       0.0         0.0  ...           NaN   \n",
       "MB-0006          0.0        0.0       0.0         0.0  ...           0.0   \n",
       "MB-0008          0.0        0.0       0.0         0.0  ...           NaN   \n",
       "...              ...        ...       ...         ...  ...           ...   \n",
       "MTS-T2428        0.0        0.0       0.0         0.0  ...           NaN   \n",
       "MTS-T2429        0.0        0.0       1.0         0.0  ...           NaN   \n",
       "MTS-T2430        0.0        0.0       0.0         0.0  ...           NaN   \n",
       "MTS-T2431        0.0        0.0       0.0         0.0  ...           NaN   \n",
       "MTS-T2432        0.0        0.0       0.0         0.0  ...           NaN   \n",
       "\n",
       "           meth_STK11  meth_TBX3  meth_TP53  meth_XRCC2  meth_ZMYM3    mut  \\\n",
       "MB-0000           NaN        NaN        NaN         NaN         NaN  False   \n",
       "MB-0002           NaN        NaN        NaN         NaN         NaN   True   \n",
       "MB-0005           NaN        NaN        NaN         NaN         NaN   True   \n",
       "MB-0006           0.0   0.079208        0.0     0.00365    0.348101   True   \n",
       "MB-0008           NaN        NaN        NaN         NaN         NaN   True   \n",
       "...               ...        ...        ...         ...         ...    ...   \n",
       "MTS-T2428         NaN        NaN        NaN         NaN         NaN   True   \n",
       "MTS-T2429         NaN        NaN        NaN         NaN         NaN   True   \n",
       "MTS-T2430         NaN        NaN        NaN         NaN         NaN   True   \n",
       "MTS-T2431         NaN        NaN        NaN         NaN         NaN   True   \n",
       "MTS-T2432         NaN        NaN        NaN         NaN         NaN   True   \n",
       "\n",
       "             cna    rna   meth  \n",
       "MB-0000     True   True  False  \n",
       "MB-0002     True   True  False  \n",
       "MB-0005     True   True  False  \n",
       "MB-0006     True   True   True  \n",
       "MB-0008     True   True  False  \n",
       "...          ...    ...    ...  \n",
       "MTS-T2428  False  False  False  \n",
       "MTS-T2429  False  False  False  \n",
       "MTS-T2430  False  False  False  \n",
       "MTS-T2431  False  False  False  \n",
       "MTS-T2432  False  False  False  \n",
       "\n",
       "[2454 rows x 274 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matrix = mut_matrix.merge(cna_matrix, left_index=True, right_index=True, how='outer')\n",
    "matrix = matrix.merge(rna_matrix, left_index=True, right_index=True, how='outer')\n",
    "matrix = matrix.merge(meth_matrix, left_index=True, right_index=True, how='outer')\n",
    "\n",
    "matrix.loc[matrix['mut'] != True, 'mut'] = False\n",
    "matrix.loc[matrix['cna'] != True, 'cna'] = False\n",
    "matrix.loc[matrix['rna'] != True, 'rna'] = False\n",
    "matrix.loc[matrix['meth'] != True, 'meth'] = False\n",
    "\n",
    "# change order of the columns\n",
    "cols = matrix.columns.tolist()\n",
    "cols_type = ['mut','cna','rna','meth']\n",
    "for i in cols_type:\n",
    "    cols.remove(i)\n",
    "cols = cols+cols_type\n",
    "\n",
    "matrix = matrix[cols]\n",
    "matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b3887fd",
   "metadata": {},
   "source": [
    "Add label indicating the tumor classification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "3ff61972",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mut_AKT1</th>\n",
       "      <th>mut_ARID1A</th>\n",
       "      <th>mut_ARID1B</th>\n",
       "      <th>mut_BAP1</th>\n",
       "      <th>mut_BRCA1</th>\n",
       "      <th>mut_BRCA2</th>\n",
       "      <th>mut_BRIP1</th>\n",
       "      <th>mut_CASP8</th>\n",
       "      <th>mut_CDH1</th>\n",
       "      <th>mut_CDKN1B</th>\n",
       "      <th>...</th>\n",
       "      <th>ER_STATUS</th>\n",
       "      <th>HER2_STATUS</th>\n",
       "      <th>PR_STATUS</th>\n",
       "      <th>COHORT</th>\n",
       "      <th>ER_IHC</th>\n",
       "      <th>HER2_SNP6</th>\n",
       "      <th>INTCLUST</th>\n",
       "      <th>CLAUDIN_SUBTYPE</th>\n",
       "      <th>THREEGENE</th>\n",
       "      <th>HISTOLOGICAL_SUBTYPE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MB-0000</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>4ER+</td>\n",
       "      <td>claudin-low</td>\n",
       "      <td>ER-/HER2-</td>\n",
       "      <td>Ductal/NST</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0002</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>4ER+</td>\n",
       "      <td>LumA</td>\n",
       "      <td>ER+/HER2- High Prolif</td>\n",
       "      <td>Ductal/NST</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0005</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>3</td>\n",
       "      <td>LumB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ductal/NST</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0006</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>9</td>\n",
       "      <td>LumB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MB-0008</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>9</td>\n",
       "      <td>LumB</td>\n",
       "      <td>ER+/HER2- High Prolif</td>\n",
       "      <td>Mixed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2428</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2429</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2430</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2431</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTS-T2432</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2454 rows × 286 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           mut_AKT1  mut_ARID1A  mut_ARID1B  mut_BAP1  mut_BRCA1  mut_BRCA2  \\\n",
       "MB-0000         NaN         NaN         NaN       NaN        NaN        NaN   \n",
       "MB-0002         0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MB-0005         0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MB-0006         0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MB-0008         0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "...             ...         ...         ...       ...        ...        ...   \n",
       "MTS-T2428       0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MTS-T2429       0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MTS-T2430       0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MTS-T2431       0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "MTS-T2432       0.0         0.0         0.0       0.0        0.0        0.0   \n",
       "\n",
       "           mut_BRIP1  mut_CASP8  mut_CDH1  mut_CDKN1B  ...  ER_STATUS  \\\n",
       "MB-0000          NaN        NaN       NaN         NaN  ...   Positive   \n",
       "MB-0002          0.0        0.0       0.0         0.0  ...   Positive   \n",
       "MB-0005          0.0        0.0       0.0         0.0  ...   Positive   \n",
       "MB-0006          0.0        0.0       0.0         0.0  ...   Positive   \n",
       "MB-0008          0.0        0.0       0.0         0.0  ...   Positive   \n",
       "...              ...        ...       ...         ...  ...        ...   \n",
       "MTS-T2428        0.0        0.0       0.0         0.0  ...   Positive   \n",
       "MTS-T2429        0.0        0.0       1.0         0.0  ...   Positive   \n",
       "MTS-T2430        0.0        0.0       0.0         0.0  ...        NaN   \n",
       "MTS-T2431        0.0        0.0       0.0         0.0  ...        NaN   \n",
       "MTS-T2432        0.0        0.0       0.0         0.0  ...        NaN   \n",
       "\n",
       "           HER2_STATUS  PR_STATUS  COHORT   ER_IHC  HER2_SNP6  INTCLUST  \\\n",
       "MB-0000       Negative   Negative     1.0  Positve    NEUTRAL      4ER+   \n",
       "MB-0002       Negative   Positive     1.0  Positve    NEUTRAL      4ER+   \n",
       "MB-0005       Negative   Positive     1.0  Positve    NEUTRAL         3   \n",
       "MB-0006       Negative   Positive     1.0  Positve    NEUTRAL         9   \n",
       "MB-0008       Negative   Positive     1.0  Positve    NEUTRAL         9   \n",
       "...                ...        ...     ...      ...        ...       ...   \n",
       "MTS-T2428          NaN        NaN     1.0  Positve        NaN       NaN   \n",
       "MTS-T2429          NaN        NaN     1.0  Positve        NaN       NaN   \n",
       "MTS-T2430          NaN        NaN     NaN      NaN        NaN       NaN   \n",
       "MTS-T2431          NaN        NaN     NaN      NaN        NaN       NaN   \n",
       "MTS-T2432          NaN        NaN     NaN      NaN        NaN       NaN   \n",
       "\n",
       "           CLAUDIN_SUBTYPE              THREEGENE  HISTOLOGICAL_SUBTYPE  \n",
       "MB-0000        claudin-low              ER-/HER2-            Ductal/NST  \n",
       "MB-0002               LumA  ER+/HER2- High Prolif            Ductal/NST  \n",
       "MB-0005               LumB                    NaN            Ductal/NST  \n",
       "MB-0006               LumB                    NaN                 Mixed  \n",
       "MB-0008               LumB  ER+/HER2- High Prolif                 Mixed  \n",
       "...                    ...                    ...                   ...  \n",
       "MTS-T2428              NaN                    NaN                   NaN  \n",
       "MTS-T2429              NaN                    NaN                   NaN  \n",
       "MTS-T2430              NaN                    NaN                   NaN  \n",
       "MTS-T2431              NaN                    NaN                   NaN  \n",
       "MTS-T2432              NaN                    NaN                   NaN  \n",
       "\n",
       "[2454 rows x 286 columns]"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matrix_clin_sample = clin_sample[['TMB_NONSYNONYMOUS','GRADE','ER_STATUS','HER2_STATUS','PR_STATUS']].copy()\n",
    "matrix_clin_sample.index = clin_sample.PATIENT_ID\n",
    "matrix = matrix.merge(matrix_clin_sample, left_index=True, right_index=True, how='inner')\n",
    "\n",
    "matrix_clin_patient = clin_patient[['COHORT','ER_IHC','HER2_SNP6','INTCLUST','CLAUDIN_SUBTYPE','THREEGENE','HISTOLOGICAL_SUBTYPE']].copy()\n",
    "matrix_clin_patient.index = clin_patient.PATIENT_ID\n",
    "matrix = matrix.merge(matrix_clin_patient, left_index=True, right_index=True, how='inner')\n",
    "\n",
    "matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b64b1ea",
   "metadata": {},
   "source": [
    "Save in .txt file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "0b2c190b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\simacri\\AppData\\Local\\Temp\\ipykernel_232\\1680300469.py:2: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  matrix['PATIENT_ID'] = matrix.index\n"
     ]
    }
   ],
   "source": [
    "# add column with patient id\n",
    "matrix['PATIENT_ID'] = matrix.index\n",
    "matrix.to_csv('data/data_processed.txt', sep='\\t', index=False)\n",
    "#pd.read_csv('data/data_final.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14561b62",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33d7be72",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c2a97597",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b9286d5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01c1f901",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c005f766",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba53766b",
   "metadata": {},
   "outputs": [],
   "source": [
    "count=0\n",
    "for i in cna.Hugo_Symbol.unique():\n",
    "    c = cna[cna['Hugo_Symbol']==i].Hugo_Symbol\n",
    "    if len(c) != 1:\n",
    "        print(cna[cna['Hugo_Symbol']==i])\n",
    "        count+=1\n",
    "count\n",
    "        #lista_duplicati.append(i)\n",
    "#len(lista_duplicati)\n",
    "\n",
    "\n",
    "list_rna_duplicates = []\n",
    "for i in rna.Hugo_Symbol.unique():\n",
    "    c = rna[rna['Hugo_Symbol']==i].Hugo_Symbol\n",
    "    if len(c) != 1:\n",
    "        list_rna_duplicates.append(i)\n",
    "len(list_rna_duplicates)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0b31b8e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "raw",
   "id": "b034e21e",
   "metadata": {},
   "source": [
    "['TMB_NONSYNONYMOUS','COHORT','GRADE',ER_STATUS','ER_IHC','HER2_STATUS','HER2_SNP6','PR_STATUS','INTCLUST','CLAUDIN_SUBTYPE','THREEGENE','HISTOLOGICAL_SUBTYPE']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ccb37f45",
   "metadata": {},
   "source": [
    "**TMB_NONSYNONYMOUS**\n",
    "total number of somatic/acquired mutations per coding area of a tumor genome (Mut/Mb)\n",
    "\n",
    "**COHORT**\n",
    "Cohort is a group of subjects who share a defining characteristic (It takes a value from 1 to 5)\n",
    "\n",
    "**GRADE**\n",
    "Determined by pathology by looking the nature of the cells, do they look aggressive or not (It takes a value from 1 to 3)\n",
    "\n",
    "**ER_STATUS**\n",
    "Cancer cells are positive or negative for estrogen receptors\n",
    "\n",
    "**ER_IHC**\n",
    "To assess if estrogen receptors are expressed on cancer cells by using immune-histochemistry (a dye used in pathology that targets specific antigen, if it is there, it will give a color, it is not there, the tissue on the slide will be colored) (positive/negative)\n",
    "\n",
    "**HER2_STATUS**\n",
    "Whether the cancer is positive or negative for HER2\n",
    "\n",
    "**HER2_SNP6**\n",
    "To assess if the cancer positive for HER2 or not by using advance molecular techniques (Type of next generation sequencing)\n",
    "\n",
    "**PR_STATUS**\n",
    "Cancer cells are positive or negative for progesterone receptors\n",
    "\n",
    "**INTCLUST**\n",
    "Molecular subtype of the cancer based on some gene expression (It takes a value from '4ER+', '3', '9', '7', '4ER-', '5', '8', '10', '1', '2', '6')\n",
    "\n",
    "**CLAUDIN_SUBTYPE**\n",
    "Tumor profiling test that helps show whether some estrogen receptor-positive (ER-positive), HER2-negative breast cancers are likely to metastasize (when breast cancer spreads to other organs). The claudin-low breast cancer subtype is defined by gene expression characteristics, most prominently: Low expression of cell–cell adhesion genes, high expression of epithelial–mesenchymal transition (EMT) genes, and stem cell-like/less differentiated gene expression patterns\n",
    "\n",
    "**THREEGENE**\n",
    "Three Gene classifier subtype It takes a value from 'ER-/HER2-', 'ER+/HER2- High Prolif', nan, 'ER+/HER2- Low Prolif','HER2+'\n",
    "\n",
    "**HISTOLOGICAL_SUBTYPE**\n",
    "Type of the cancer based on microscopic examination of the cancer tissue (It takes a value of 'Ductal/NST', 'Mixed', 'Lobular', 'Tubular/ cribriform', 'Mucinous', 'Medullary', 'Other', 'Metaplastic' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf745488",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "default_env_kernel",
   "language": "python",
   "name": "default_env_kernel"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
