{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "224cc1f3",
   "metadata": {},
   "source": [
    "#### Contents:\n",
    "* [Clinical and genomic data of breast cancer patients](#1)\n",
    "    * [Clinical data](#1.a)\n",
    "    * [Genomic and epigenomic data](#1.b)\n",
    "* [Genetic alterations in breast cancer](#2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d112d79b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a17b8c99",
   "metadata": {},
   "source": [
    "## Clinical and genomic data of breast cancer patients <a class=\"anchor\" id=\"1\"></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d87b239",
   "metadata": {},
   "source": [
    "### Clinical data <a class=\"anchor\" id=\"1.a\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "12038b1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "clin_patient = pd.read_csv('data/data_clinical_patient.txt', sep='\\t', comment='#')\n",
    "clin_sample = pd.read_csv('data/data_clinical_sample.txt', sep='\\t', comment='#')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "2c35b30f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['PATIENT_ID', 'LYMPH_NODES_EXAMINED_POSITIVE', 'NPI', 'CELLULARITY',\n",
      "       'CHEMOTHERAPY', 'COHORT', 'ER_IHC', 'HER2_SNP6', 'HORMONE_THERAPY',\n",
      "       'INFERRED_MENOPAUSAL_STATE', 'SEX', 'INTCLUST', 'AGE_AT_DIAGNOSIS',\n",
      "       'OS_MONTHS', 'OS_STATUS', 'CLAUDIN_SUBTYPE', 'THREEGENE',\n",
      "       'VITAL_STATUS', 'LATERALITY', 'RADIO_THERAPY', 'HISTOLOGICAL_SUBTYPE',\n",
      "       'BREAST_SURGERY', 'RFS_STATUS', 'RFS_MONTHS'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>LYMPH_NODES_EXAMINED_POSITIVE</th>\n",
       "      <th>NPI</th>\n",
       "      <th>CELLULARITY</th>\n",
       "      <th>CHEMOTHERAPY</th>\n",
       "      <th>COHORT</th>\n",
       "      <th>ER_IHC</th>\n",
       "      <th>HER2_SNP6</th>\n",
       "      <th>HORMONE_THERAPY</th>\n",
       "      <th>INFERRED_MENOPAUSAL_STATE</th>\n",
       "      <th>...</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>CLAUDIN_SUBTYPE</th>\n",
       "      <th>THREEGENE</th>\n",
       "      <th>VITAL_STATUS</th>\n",
       "      <th>LATERALITY</th>\n",
       "      <th>RADIO_THERAPY</th>\n",
       "      <th>HISTOLOGICAL_SUBTYPE</th>\n",
       "      <th>BREAST_SURGERY</th>\n",
       "      <th>RFS_STATUS</th>\n",
       "      <th>RFS_MONTHS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MB-0000</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6.044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NO</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>YES</td>\n",
       "      <td>Post</td>\n",
       "      <td>...</td>\n",
       "      <td>0:LIVING</td>\n",
       "      <td>claudin-low</td>\n",
       "      <td>ER-/HER2-</td>\n",
       "      <td>Living</td>\n",
       "      <td>Right</td>\n",
       "      <td>YES</td>\n",
       "      <td>Ductal/NST</td>\n",
       "      <td>MASTECTOMY</td>\n",
       "      <td>0:Not Recurred</td>\n",
       "      <td>138.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MB-0002</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.020</td>\n",
       "      <td>High</td>\n",
       "      <td>NO</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>YES</td>\n",
       "      <td>Pre</td>\n",
       "      <td>...</td>\n",
       "      <td>0:LIVING</td>\n",
       "      <td>LumA</td>\n",
       "      <td>ER+/HER2- High Prolif</td>\n",
       "      <td>Living</td>\n",
       "      <td>Right</td>\n",
       "      <td>YES</td>\n",
       "      <td>Ductal/NST</td>\n",
       "      <td>BREAST CONSERVING</td>\n",
       "      <td>0:Not Recurred</td>\n",
       "      <td>83.52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MB-0005</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.030</td>\n",
       "      <td>High</td>\n",
       "      <td>YES</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>YES</td>\n",
       "      <td>Pre</td>\n",
       "      <td>...</td>\n",
       "      <td>1:DECEASED</td>\n",
       "      <td>LumB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Died of Disease</td>\n",
       "      <td>Right</td>\n",
       "      <td>NO</td>\n",
       "      <td>Ductal/NST</td>\n",
       "      <td>MASTECTOMY</td>\n",
       "      <td>1:Recurred</td>\n",
       "      <td>151.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MB-0006</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.050</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>YES</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>YES</td>\n",
       "      <td>Pre</td>\n",
       "      <td>...</td>\n",
       "      <td>0:LIVING</td>\n",
       "      <td>LumB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Living</td>\n",
       "      <td>Right</td>\n",
       "      <td>YES</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>MASTECTOMY</td>\n",
       "      <td>0:Not Recurred</td>\n",
       "      <td>162.76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB-0008</td>\n",
       "      <td>8.0</td>\n",
       "      <td>6.080</td>\n",
       "      <td>High</td>\n",
       "      <td>YES</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>YES</td>\n",
       "      <td>Post</td>\n",
       "      <td>...</td>\n",
       "      <td>1:DECEASED</td>\n",
       "      <td>LumB</td>\n",
       "      <td>ER+/HER2- High Prolif</td>\n",
       "      <td>Died of Disease</td>\n",
       "      <td>Right</td>\n",
       "      <td>YES</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>MASTECTOMY</td>\n",
       "      <td>1:Recurred</td>\n",
       "      <td>18.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2504</th>\n",
       "      <td>MTS-T2428</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:Recurred</td>\n",
       "      <td>4.93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2505</th>\n",
       "      <td>MTS-T2429</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.560</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Positve</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:Recurred</td>\n",
       "      <td>16.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2506</th>\n",
       "      <td>MTS-T2430</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2507</th>\n",
       "      <td>MTS-T2431</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2508</th>\n",
       "      <td>MTS-T2432</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2509 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     PATIENT_ID  LYMPH_NODES_EXAMINED_POSITIVE    NPI CELLULARITY  \\\n",
       "0       MB-0000                           10.0  6.044         NaN   \n",
       "1       MB-0002                            0.0  4.020        High   \n",
       "2       MB-0005                            1.0  4.030        High   \n",
       "3       MB-0006                            3.0  4.050    Moderate   \n",
       "4       MB-0008                            8.0  6.080        High   \n",
       "...         ...                            ...    ...         ...   \n",
       "2504  MTS-T2428                            0.0  2.540         NaN   \n",
       "2505  MTS-T2429                            0.0  4.560         NaN   \n",
       "2506  MTS-T2430                            0.0    NaN         NaN   \n",
       "2507  MTS-T2431                            0.0    NaN         NaN   \n",
       "2508  MTS-T2432                            0.0    NaN         NaN   \n",
       "\n",
       "     CHEMOTHERAPY  COHORT   ER_IHC HER2_SNP6 HORMONE_THERAPY  \\\n",
       "0              NO     1.0  Positve   NEUTRAL             YES   \n",
       "1              NO     1.0  Positve   NEUTRAL             YES   \n",
       "2             YES     1.0  Positve   NEUTRAL             YES   \n",
       "3             YES     1.0  Positve   NEUTRAL             YES   \n",
       "4             YES     1.0  Positve   NEUTRAL             YES   \n",
       "...           ...     ...      ...       ...             ...   \n",
       "2504          NaN     1.0  Positve       NaN             NaN   \n",
       "2505          NaN     1.0  Positve       NaN             NaN   \n",
       "2506          NaN     NaN      NaN       NaN             NaN   \n",
       "2507          NaN     NaN      NaN       NaN             NaN   \n",
       "2508          NaN     NaN      NaN       NaN             NaN   \n",
       "\n",
       "     INFERRED_MENOPAUSAL_STATE  ...   OS_STATUS CLAUDIN_SUBTYPE  \\\n",
       "0                         Post  ...    0:LIVING     claudin-low   \n",
       "1                          Pre  ...    0:LIVING            LumA   \n",
       "2                          Pre  ...  1:DECEASED            LumB   \n",
       "3                          Pre  ...    0:LIVING            LumB   \n",
       "4                         Post  ...  1:DECEASED            LumB   \n",
       "...                        ...  ...         ...             ...   \n",
       "2504                       NaN  ...         NaN             NaN   \n",
       "2505                       NaN  ...         NaN             NaN   \n",
       "2506                       NaN  ...         NaN             NaN   \n",
       "2507                       NaN  ...         NaN             NaN   \n",
       "2508                       NaN  ...         NaN             NaN   \n",
       "\n",
       "                  THREEGENE     VITAL_STATUS LATERALITY RADIO_THERAPY  \\\n",
       "0                 ER-/HER2-           Living      Right           YES   \n",
       "1     ER+/HER2- High Prolif           Living      Right           YES   \n",
       "2                       NaN  Died of Disease      Right            NO   \n",
       "3                       NaN           Living      Right           YES   \n",
       "4     ER+/HER2- High Prolif  Died of Disease      Right           YES   \n",
       "...                     ...              ...        ...           ...   \n",
       "2504                    NaN              NaN        NaN           NaN   \n",
       "2505                    NaN              NaN        NaN           NaN   \n",
       "2506                    NaN              NaN        NaN           NaN   \n",
       "2507                    NaN              NaN        NaN           NaN   \n",
       "2508                    NaN              NaN        NaN           NaN   \n",
       "\n",
       "     HISTOLOGICAL_SUBTYPE     BREAST_SURGERY      RFS_STATUS RFS_MONTHS  \n",
       "0              Ductal/NST         MASTECTOMY  0:Not Recurred     138.65  \n",
       "1              Ductal/NST  BREAST CONSERVING  0:Not Recurred      83.52  \n",
       "2              Ductal/NST         MASTECTOMY      1:Recurred     151.28  \n",
       "3                   Mixed         MASTECTOMY  0:Not Recurred     162.76  \n",
       "4                   Mixed         MASTECTOMY      1:Recurred      18.55  \n",
       "...                   ...                ...             ...        ...  \n",
       "2504                  NaN                NaN      1:Recurred       4.93  \n",
       "2505                  NaN                NaN      1:Recurred      16.18  \n",
       "2506                  NaN                NaN             NaN        NaN  \n",
       "2507                  NaN                NaN             NaN        NaN  \n",
       "2508                  NaN                NaN             NaN        NaN  \n",
       "\n",
       "[2509 rows x 24 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(clin_patient.columns)\n",
    "clin_patient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "91f6b5c5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['PATIENT_ID', 'SAMPLE_ID', 'CANCER_TYPE', 'CANCER_TYPE_DETAILED',\n",
      "       'ER_STATUS', 'HER2_STATUS', 'GRADE', 'ONCOTREE_CODE', 'PR_STATUS',\n",
      "       'SAMPLE_TYPE', 'TUMOR_SIZE', 'TUMOR_STAGE', 'TMB_NONSYNONYMOUS'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>CANCER_TYPE</th>\n",
       "      <th>CANCER_TYPE_DETAILED</th>\n",
       "      <th>ER_STATUS</th>\n",
       "      <th>HER2_STATUS</th>\n",
       "      <th>GRADE</th>\n",
       "      <th>ONCOTREE_CODE</th>\n",
       "      <th>PR_STATUS</th>\n",
       "      <th>SAMPLE_TYPE</th>\n",
       "      <th>TUMOR_SIZE</th>\n",
       "      <th>TUMOR_STAGE</th>\n",
       "      <th>TMB_NONSYNONYMOUS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MB-0000</td>\n",
       "      <td>MB-0000</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>3.0</td>\n",
       "      <td>IDC</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Primary</td>\n",
       "      <td>22.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MB-0002</td>\n",
       "      <td>MB-0002</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>3.0</td>\n",
       "      <td>IDC</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Primary</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.615035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MB-0005</td>\n",
       "      <td>MB-0005</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>2.0</td>\n",
       "      <td>IDC</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Primary</td>\n",
       "      <td>15.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.615035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MB-0006</td>\n",
       "      <td>MB-0006</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Mixed Ductal and Lobular Carcinoma</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>2.0</td>\n",
       "      <td>MDLC</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Primary</td>\n",
       "      <td>25.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.307518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB-0008</td>\n",
       "      <td>MB-0008</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Mixed Ductal and Lobular Carcinoma</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>3.0</td>\n",
       "      <td>MDLC</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Primary</td>\n",
       "      <td>40.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.615035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2504</th>\n",
       "      <td>MTS-T2428</td>\n",
       "      <td>MTS-T2428</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>Positive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.615035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2505</th>\n",
       "      <td>MTS-T2429</td>\n",
       "      <td>MTS-T2429</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>Positive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>28.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.230071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2506</th>\n",
       "      <td>MTS-T2430</td>\n",
       "      <td>MTS-T2430</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.845106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2507</th>\n",
       "      <td>MTS-T2431</td>\n",
       "      <td>MTS-T2431</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.152624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2508</th>\n",
       "      <td>MTS-T2432</td>\n",
       "      <td>MTS-T2432</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BRCA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.537589</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2509 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     PATIENT_ID  SAMPLE_ID    CANCER_TYPE  \\\n",
       "0       MB-0000    MB-0000  Breast Cancer   \n",
       "1       MB-0002    MB-0002  Breast Cancer   \n",
       "2       MB-0005    MB-0005  Breast Cancer   \n",
       "3       MB-0006    MB-0006  Breast Cancer   \n",
       "4       MB-0008    MB-0008  Breast Cancer   \n",
       "...         ...        ...            ...   \n",
       "2504  MTS-T2428  MTS-T2428  Breast Cancer   \n",
       "2505  MTS-T2429  MTS-T2429  Breast Cancer   \n",
       "2506  MTS-T2430  MTS-T2430  Breast Cancer   \n",
       "2507  MTS-T2431  MTS-T2431  Breast Cancer   \n",
       "2508  MTS-T2432  MTS-T2432  Breast Cancer   \n",
       "\n",
       "                           CANCER_TYPE_DETAILED ER_STATUS HER2_STATUS  GRADE  \\\n",
       "0              Breast Invasive Ductal Carcinoma  Positive    Negative    3.0   \n",
       "1              Breast Invasive Ductal Carcinoma  Positive    Negative    3.0   \n",
       "2              Breast Invasive Ductal Carcinoma  Positive    Negative    2.0   \n",
       "3     Breast Mixed Ductal and Lobular Carcinoma  Positive    Negative    2.0   \n",
       "4     Breast Mixed Ductal and Lobular Carcinoma  Positive    Negative    3.0   \n",
       "...                                         ...       ...         ...    ...   \n",
       "2504                  Invasive Breast Carcinoma  Positive         NaN    1.0   \n",
       "2505                  Invasive Breast Carcinoma  Positive         NaN    2.0   \n",
       "2506                  Invasive Breast Carcinoma       NaN         NaN    NaN   \n",
       "2507                  Invasive Breast Carcinoma       NaN         NaN    NaN   \n",
       "2508                  Invasive Breast Carcinoma       NaN         NaN    NaN   \n",
       "\n",
       "     ONCOTREE_CODE PR_STATUS SAMPLE_TYPE  TUMOR_SIZE  TUMOR_STAGE  \\\n",
       "0              IDC  Negative     Primary        22.0          2.0   \n",
       "1              IDC  Positive     Primary        10.0          1.0   \n",
       "2              IDC  Positive     Primary        15.0          2.0   \n",
       "3             MDLC  Positive     Primary        25.0          2.0   \n",
       "4             MDLC  Positive     Primary        40.0          2.0   \n",
       "...            ...       ...         ...         ...          ...   \n",
       "2504          BRCA       NaN     Primary        27.0          1.0   \n",
       "2505          BRCA       NaN     Primary        28.0          2.0   \n",
       "2506          BRCA       NaN     Primary         NaN          0.0   \n",
       "2507          BRCA       NaN     Primary         NaN          0.0   \n",
       "2508          BRCA       NaN     Primary         NaN          0.0   \n",
       "\n",
       "      TMB_NONSYNONYMOUS  \n",
       "0              0.000000  \n",
       "1              2.615035  \n",
       "2              2.615035  \n",
       "3              1.307518  \n",
       "4              2.615035  \n",
       "...                 ...  \n",
       "2504           2.615035  \n",
       "2505           5.230071  \n",
       "2506           7.845106  \n",
       "2507           9.152624  \n",
       "2508           6.537589  \n",
       "\n",
       "[2509 rows x 13 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(clin_sample.columns)\n",
    "clin_sample"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d7a9d3a0",
   "metadata": {},
   "source": [
    "### Genomic and epigenomic data  <a class=\"anchor\" id=\"1.b\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "49ef012a",
   "metadata": {},
   "outputs": [],
   "source": [
    "mut = pd.read_csv('data/data_mutations.txt', sep='\\t', comment='#')\n",
    "cna = pd.read_csv('data/data_cna.txt', sep='\\t')\n",
    "rna = pd.read_csv('data/data_mrna_illumina_microarray_zscores_ref_diploid_samples.txt', sep='\\t', comment='#')\n",
    "meth = pd.read_csv('data/data_methylation_promoters_rrbs.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fdaa079b",
   "metadata": {},
   "source": [
    "#### small mutations"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4e66b43",
   "metadata": {},
   "source": [
    "The **mut** dataframe stores 17272 small variants of length ranging from one nucleotide to few tens of nucleotides. Each variant is associated to a patient and is a variation of a DNA fraction, when compared with the reference genome. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "196cbe67",
   "metadata": {},
   "source": [
    "The most relevant informations are stored in the following columns:  \n",
    "**Hugo_Symbol**: identification symbol of the gene in which the variant occurs  \n",
    "**Chromosome**, **Start_Position**, **End_Position** and **strand**: exact position of the variant in the DNA molecule  \n",
    "**Consequence**kind : effect of the variant presence\n",
    "**Variant_Classification** and **Variant_Type**: type of mutation (insertion, deletion, single nucleotide variant...)  \n",
    "**Reference_Allele**: nucleotide(s) found in the reference genome at the same variant location  \n",
    "**Tumor_Seq_Allele1** and **Tumor_Seq_Allele2**: mutation found in each allele (in present in both)\n",
    "**Tumor_Sample_Barcode**: sample id (referred to the patient)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "7773e312",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Hugo_Symbol', 'Entrez_Gene_Id', 'Center', 'NCBI_Build', 'Chromosome',\n",
      "       'Start_Position', 'End_Position', 'Strand', 'Consequence',\n",
      "       'Variant_Classification', 'Variant_Type', 'Reference_Allele',\n",
      "       'Tumor_Seq_Allele1', 'Tumor_Seq_Allele2', 'dbSNP_RS',\n",
      "       'dbSNP_Val_Status', 'Tumor_Sample_Barcode',\n",
      "       'Matched_Norm_Sample_Barcode', 'Match_Norm_Seq_Allele1',\n",
      "       'Match_Norm_Seq_Allele2', 'Tumor_Validation_Allele1',\n",
      "       'Tumor_Validation_Allele2', 'Match_Norm_Validation_Allele1',\n",
      "       'Match_Norm_Validation_Allele2', 'Verification_Status',\n",
      "       'Validation_Status', 'Mutation_Status', 'Sequencing_Phase',\n",
      "       'Sequence_Source', 'Validation_Method', 'Score', 'BAM_File',\n",
      "       'Sequencer', 't_ref_count', 't_alt_count', 'n_ref_count', 'n_alt_count',\n",
      "       'HGVSc', 'HGVSp', 'HGVSp_Short', 'Transcript_ID', 'RefSeq',\n",
      "       'Protein_position', 'Codons', 'Hotspot'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>...</th>\n",
       "      <th>n_ref_count</th>\n",
       "      <th>n_alt_count</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>RefSeq</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Hotspot</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TP53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>METABRIC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7579344</td>\n",
       "      <td>7579345</td>\n",
       "      <td>+</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000269305.4:c.343dup</td>\n",
       "      <td>p.His115ProfsTer34</td>\n",
       "      <td>p.H115Pfs*34</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>114.0</td>\n",
       "      <td>-/C</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TP53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>METABRIC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7579346</td>\n",
       "      <td>7579347</td>\n",
       "      <td>+</td>\n",
       "      <td>protein_altering_variant</td>\n",
       "      <td>In_Frame_Ins</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000269305.4:c.340_341insCTG</td>\n",
       "      <td>p.Leu114delinsSerVal</td>\n",
       "      <td>p.L114delinsSV</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>114.0</td>\n",
       "      <td>ttg/tCTGtg</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MLLT4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>METABRIC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>6</td>\n",
       "      <td>168299111</td>\n",
       "      <td>168299111</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000392108.3:c.1544G&gt;T</td>\n",
       "      <td>p.Gly515Val</td>\n",
       "      <td>p.G515V</td>\n",
       "      <td>ENST00000392108</td>\n",
       "      <td>NM_001040000.2</td>\n",
       "      <td>515.0</td>\n",
       "      <td>gGa/gTa</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NF2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>METABRIC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>22</td>\n",
       "      <td>29999995</td>\n",
       "      <td>29999995</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000338641.4:c.8G&gt;T</td>\n",
       "      <td>p.Gly3Val</td>\n",
       "      <td>p.G3V</td>\n",
       "      <td>ENST00000338641</td>\n",
       "      <td>NM_000268.3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>gGg/gTg</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SF3B1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>METABRIC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>2</td>\n",
       "      <td>198288682</td>\n",
       "      <td>198288682</td>\n",
       "      <td>+</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Silent</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000335508.6:c.45T&gt;A</td>\n",
       "      <td>p.Ile15=</td>\n",
       "      <td>p.I15=</td>\n",
       "      <td>ENST00000335508</td>\n",
       "      <td>NM_012433.2</td>\n",
       "      <td>15.0</td>\n",
       "      <td>atT/atA</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17267</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>METABRIC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>3</td>\n",
       "      <td>178917478</td>\n",
       "      <td>178917478</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000263967.3:c.353G&gt;A</td>\n",
       "      <td>p.Gly118Asp</td>\n",
       "      <td>p.G118D</td>\n",
       "      <td>ENST00000263967</td>\n",
       "      <td>NM_006218.2</td>\n",
       "      <td>118.0</td>\n",
       "      <td>gGt/gAt</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17268</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>METABRIC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>3</td>\n",
       "      <td>178952085</td>\n",
       "      <td>178952085</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000263967.3:c.3140A&gt;G</td>\n",
       "      <td>p.His1047Arg</td>\n",
       "      <td>p.H1047R</td>\n",
       "      <td>ENST00000263967</td>\n",
       "      <td>NM_006218.2</td>\n",
       "      <td>1047.0</td>\n",
       "      <td>cAt/cGt</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17269</th>\n",
       "      <td>TP53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>METABRIC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7579355</td>\n",
       "      <td>7579355</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000269305.4:c.332T&gt;C</td>\n",
       "      <td>p.Leu111Pro</td>\n",
       "      <td>p.L111P</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>111.0</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17270</th>\n",
       "      <td>JAK1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>METABRIC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>65311232</td>\n",
       "      <td>65311233</td>\n",
       "      <td>+</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000342505.4:c.2079dup</td>\n",
       "      <td>p.Val694SerfsTer15</td>\n",
       "      <td>p.V694Sfs*15</td>\n",
       "      <td>ENST00000342505</td>\n",
       "      <td>NM_002227.2</td>\n",
       "      <td>693.0</td>\n",
       "      <td>aaa/aaAa</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17271</th>\n",
       "      <td>DNAH11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>METABRIC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>7</td>\n",
       "      <td>21627709</td>\n",
       "      <td>21627709</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000328843.6:c.1738G&gt;C</td>\n",
       "      <td>p.Glu580Gln</td>\n",
       "      <td>p.E580Q</td>\n",
       "      <td>ENST00000328843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>580.0</td>\n",
       "      <td>Gag/Cag</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17272 rows × 45 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Hugo_Symbol  Entrez_Gene_Id    Center NCBI_Build Chromosome  \\\n",
       "0            TP53             NaN  METABRIC     GRCh37         17   \n",
       "1            TP53             NaN  METABRIC     GRCh37         17   \n",
       "2           MLLT4             NaN  METABRIC     GRCh37          6   \n",
       "3             NF2             NaN  METABRIC     GRCh37         22   \n",
       "4           SF3B1             NaN  METABRIC     GRCh37          2   \n",
       "...           ...             ...       ...        ...        ...   \n",
       "17267      PIK3CA             NaN  METABRIC     GRCh37          3   \n",
       "17268      PIK3CA             NaN  METABRIC     GRCh37          3   \n",
       "17269        TP53             NaN  METABRIC     GRCh37         17   \n",
       "17270        JAK1             NaN  METABRIC     GRCh37          1   \n",
       "17271      DNAH11             NaN  METABRIC     GRCh37          7   \n",
       "\n",
       "       Start_Position  End_Position Strand  \\\n",
       "0             7579344       7579345      +   \n",
       "1             7579346       7579347      +   \n",
       "2           168299111     168299111      +   \n",
       "3            29999995      29999995      +   \n",
       "4           198288682     198288682      +   \n",
       "...               ...           ...    ...   \n",
       "17267       178917478     178917478      +   \n",
       "17268       178952085     178952085      +   \n",
       "17269         7579355       7579355      +   \n",
       "17270        65311232      65311233      +   \n",
       "17271        21627709      21627709      +   \n",
       "\n",
       "                                  Consequence Variant_Classification  ...  \\\n",
       "0                          frameshift_variant        Frame_Shift_Ins  ...   \n",
       "1                    protein_altering_variant           In_Frame_Ins  ...   \n",
       "2                            missense_variant      Missense_Mutation  ...   \n",
       "3                            missense_variant      Missense_Mutation  ...   \n",
       "4                          synonymous_variant                 Silent  ...   \n",
       "...                                       ...                    ...  ...   \n",
       "17267  missense_variant,splice_region_variant      Missense_Mutation  ...   \n",
       "17268                        missense_variant      Missense_Mutation  ...   \n",
       "17269                        missense_variant      Missense_Mutation  ...   \n",
       "17270                      frameshift_variant        Frame_Shift_Ins  ...   \n",
       "17271                        missense_variant      Missense_Mutation  ...   \n",
       "\n",
       "      n_ref_count n_alt_count                              HGVSc  \\\n",
       "0             NaN         NaN         ENST00000269305.4:c.343dup   \n",
       "1             NaN         NaN  ENST00000269305.4:c.340_341insCTG   \n",
       "2             NaN         NaN        ENST00000392108.3:c.1544G>T   \n",
       "3             NaN         NaN           ENST00000338641.4:c.8G>T   \n",
       "4             NaN         NaN          ENST00000335508.6:c.45T>A   \n",
       "...           ...         ...                                ...   \n",
       "17267         NaN         NaN         ENST00000263967.3:c.353G>A   \n",
       "17268         NaN         NaN        ENST00000263967.3:c.3140A>G   \n",
       "17269         NaN         NaN         ENST00000269305.4:c.332T>C   \n",
       "17270         NaN         NaN        ENST00000342505.4:c.2079dup   \n",
       "17271         NaN         NaN        ENST00000328843.6:c.1738G>C   \n",
       "\n",
       "                      HGVSp     HGVSp_Short    Transcript_ID          RefSeq  \\\n",
       "0        p.His115ProfsTer34    p.H115Pfs*34  ENST00000269305  NM_001126112.2   \n",
       "1      p.Leu114delinsSerVal  p.L114delinsSV  ENST00000269305  NM_001126112.2   \n",
       "2               p.Gly515Val         p.G515V  ENST00000392108  NM_001040000.2   \n",
       "3                 p.Gly3Val           p.G3V  ENST00000338641     NM_000268.3   \n",
       "4                  p.Ile15=          p.I15=  ENST00000335508     NM_012433.2   \n",
       "...                     ...             ...              ...             ...   \n",
       "17267           p.Gly118Asp         p.G118D  ENST00000263967     NM_006218.2   \n",
       "17268          p.His1047Arg        p.H1047R  ENST00000263967     NM_006218.2   \n",
       "17269           p.Leu111Pro         p.L111P  ENST00000269305  NM_001126112.2   \n",
       "17270    p.Val694SerfsTer15    p.V694Sfs*15  ENST00000342505     NM_002227.2   \n",
       "17271           p.Glu580Gln         p.E580Q  ENST00000328843             NaN   \n",
       "\n",
       "       Protein_position      Codons  Hotspot  \n",
       "0                 114.0         -/C        0  \n",
       "1                 114.0  ttg/tCTGtg        0  \n",
       "2                 515.0     gGa/gTa        0  \n",
       "3                   3.0     gGg/gTg        0  \n",
       "4                  15.0     atT/atA        0  \n",
       "...                 ...         ...      ...  \n",
       "17267             118.0     gGt/gAt        0  \n",
       "17268            1047.0     cAt/cGt        0  \n",
       "17269             111.0     cTg/cCg        0  \n",
       "17270             693.0    aaa/aaAa        0  \n",
       "17271             580.0     Gag/Cag        0  \n",
       "\n",
       "[17272 rows x 45 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(mut.columns)\n",
    "mut"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ddf6aeed",
   "metadata": {},
   "source": [
    "#### copy number variations"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d31a17c6",
   "metadata": {},
   "source": [
    "The **cna** dataframe stores, for each patient, copy number alterations (replications or deletions of large DNA fractions) referred to 22544 genes. Rows represent genes and columns represent patients.   \n",
    "Values:  \n",
    "-2 = homozygous deletion  \n",
    "-1 = hemizygous deletion  \n",
    "0 = neutral / no change  \n",
    "1 = gain  \n",
    "2 = high level amplification.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "b06162db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of genes with copy number variations:  22542\n",
      "Number of patients with copy number variations:  2173\n"
     ]
    }
   ],
   "source": [
    "print('Total number of genes with copy number variations: ',len(cna.Hugo_Symbol.unique()))\n",
    "print('Number of patients with copy number variations: ',len(cna.columns)-2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "8b897e59",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>MB-0000</th>\n",
       "      <th>MB-0039</th>\n",
       "      <th>MB-0045</th>\n",
       "      <th>MB-0046</th>\n",
       "      <th>MB-0048</th>\n",
       "      <th>MB-0050</th>\n",
       "      <th>MB-0053</th>\n",
       "      <th>MB-0062</th>\n",
       "      <th>...</th>\n",
       "      <th>MB-5467</th>\n",
       "      <th>MB-5546</th>\n",
       "      <th>MB-5585</th>\n",
       "      <th>MB-5625</th>\n",
       "      <th>MB-5648</th>\n",
       "      <th>MB-6020</th>\n",
       "      <th>MB-6213</th>\n",
       "      <th>MB-6230</th>\n",
       "      <th>MB-7148</th>\n",
       "      <th>MB-7188</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>503538.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M-AS1</td>\n",
       "      <td>144571.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22539</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22540</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>79699.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22541</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>7791.0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22542</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-2</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22543</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22544 rows × 2175 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Hugo_Symbol  Entrez_Gene_Id  MB-0000  MB-0039  MB-0045  MB-0046  \\\n",
       "0            A1BG             1.0        0        0       -1        0   \n",
       "1        A1BG-AS1        503538.0        0        0       -1        0   \n",
       "2            A1CF         29974.0        0        0        0        0   \n",
       "3             A2M             2.0        0        0       -1       -1   \n",
       "4         A2M-AS1        144571.0        0        0       -1       -1   \n",
       "...           ...             ...      ...      ...      ...      ...   \n",
       "22539      ZYG11A             NaN        0        0        1        0   \n",
       "22540      ZYG11B         79699.0        0        0       -1        0   \n",
       "22541         ZYX          7791.0        0       -1        0        0   \n",
       "22542       ZZEF1         23140.0        0        0       -2       -1   \n",
       "22543        ZZZ3         26009.0        0        0        0        0   \n",
       "\n",
       "       MB-0048  MB-0050  MB-0053  MB-0062  ...  MB-5467  MB-5546  MB-5585  \\\n",
       "0            0        0        0       -1  ...      0.0      0.0      0.0   \n",
       "1            0        0        0       -1  ...      0.0      0.0      0.0   \n",
       "2            1        0        0        0  ...      0.0      0.0      0.0   \n",
       "3            0        0        0        2  ...      0.0      0.0      0.0   \n",
       "4            0        0        0        2  ...      0.0      0.0      0.0   \n",
       "...        ...      ...      ...      ...  ...      ...      ...      ...   \n",
       "22539        0        0        1        1  ...      1.0      0.0     -1.0   \n",
       "22540        0        0        1        1  ...      1.0      0.0     -1.0   \n",
       "22541        1        0        0        0  ...      0.0      0.0      0.0   \n",
       "22542       -1       -1        0       -1  ...      0.0      0.0      0.0   \n",
       "22543        0        0        0        1  ...      0.0      0.0     -1.0   \n",
       "\n",
       "       MB-5625  MB-5648  MB-6020  MB-6213  MB-6230  MB-7148  MB-7188  \n",
       "0          0.0      0.0      1.0      0.0      0.0      0.0      0.0  \n",
       "1          0.0      0.0      1.0      0.0      0.0      0.0      0.0  \n",
       "2          0.0      0.0      0.0      0.0      0.0     -1.0      0.0  \n",
       "3          0.0      0.0      0.0      1.0      0.0      0.0      0.0  \n",
       "4          0.0      0.0      0.0      1.0      0.0      0.0      0.0  \n",
       "...        ...      ...      ...      ...      ...      ...      ...  \n",
       "22539      0.0     -1.0      0.0      0.0     -1.0      0.0      0.0  \n",
       "22540      0.0     -1.0      0.0      0.0     -1.0      0.0      0.0  \n",
       "22541      0.0      1.0      0.0      0.0      0.0      0.0     -1.0  \n",
       "22542      0.0      0.0     -1.0      0.0      0.0      0.0      0.0  \n",
       "22543      0.0     -1.0      2.0      0.0      0.0      0.0      0.0  \n",
       "\n",
       "[22544 rows x 2175 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "f0e9f3d3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Hugo_Symbol  Entrez_Gene_Id  MB-0000  MB-0039  MB-0045  MB-0046  \\\n",
      "509    PALM2AKAP2        445815.0        0        0       -1        0   \n",
      "14303  PALM2AKAP2        445815.0        0        0        0        0   \n",
      "\n",
      "       MB-0048  MB-0050  MB-0053  MB-0062  ...  MB-5467  MB-5546  MB-5585  \\\n",
      "509          0        0       -1       -1  ...      0.0      0.0      0.0   \n",
      "14303        0        0        0       -1  ...      0.0      0.0      0.0   \n",
      "\n",
      "       MB-5625  MB-5648  MB-6020  MB-6213  MB-6230  MB-7148  MB-7188  \n",
      "509        0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
      "14303      0.0      0.0      0.0      0.0      0.0      0.0      0.0  \n",
      "\n",
      "[2 rows x 2175 columns]\n",
      "     Hugo_Symbol  Entrez_Gene_Id  MB-0000  MB-0039  MB-0045  MB-0046  MB-0048  \\\n",
      "2358     CACNA1C           775.0        0        0       -1       -1       -1   \n",
      "2362     CACNA1C             NaN        0        0       -1       -1       -1   \n",
      "\n",
      "      MB-0050  MB-0053  MB-0062  ...  MB-5467  MB-5546  MB-5585  MB-5625  \\\n",
      "2358       -1        0        2  ...      0.0      0.0      0.0     -2.0   \n",
      "2362       -1        0        2  ...      0.0      0.0      0.0      0.0   \n",
      "\n",
      "      MB-5648  MB-6020  MB-6213  MB-6230  MB-7148  MB-7188  \n",
      "2358      0.0      0.0      1.0      0.0      0.0      0.0  \n",
      "2362      0.0      0.0      1.0      0.0      0.0      0.0  \n",
      "\n",
      "[2 rows x 2175 columns]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count=0\n",
    "for i in cna.Hugo_Symbol.unique():\n",
    "    c = cna[cna['Hugo_Symbol']==i].Hugo_Symbol\n",
    "    if len(c) != 1:\n",
    "        print(cna[cna['Hugo_Symbol']==i])\n",
    "        count+=1\n",
    "count\n",
    "        #lista_duplicati.append(i)\n",
    "#len(lista_duplicati)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ee62a822",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "6b38096c",
   "metadata": {},
   "source": [
    "#### rna expression"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e67ec77",
   "metadata": {},
   "source": [
    "The **rna** dataframe stores, for every patients (columns), the mRNA expression values of 20603 genes (rows)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "b2a91b0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of genes:  20387\n",
      "Number of patients:  1980\n"
     ]
    }
   ],
   "source": [
    "print('Total number of genes: ',len(rna.Hugo_Symbol.unique()))\n",
    "print('Number of patients: ',len(rna.columns)-2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2554af22",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>MB-0362</th>\n",
       "      <th>MB-0346</th>\n",
       "      <th>MB-0386</th>\n",
       "      <th>MB-0574</th>\n",
       "      <th>MB-0185</th>\n",
       "      <th>MB-0503</th>\n",
       "      <th>MB-0641</th>\n",
       "      <th>MB-0201</th>\n",
       "      <th>...</th>\n",
       "      <th>MB-6192</th>\n",
       "      <th>MB-4820</th>\n",
       "      <th>MB-5527</th>\n",
       "      <th>MB-5167</th>\n",
       "      <th>MB-5465</th>\n",
       "      <th>MB-5453</th>\n",
       "      <th>MB-5471</th>\n",
       "      <th>MB-5127</th>\n",
       "      <th>MB-4313</th>\n",
       "      <th>MB-4823</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RERE</td>\n",
       "      <td>473</td>\n",
       "      <td>-0.7139</td>\n",
       "      <td>1.2266</td>\n",
       "      <td>-0.0053</td>\n",
       "      <td>-0.4399</td>\n",
       "      <td>-0.5958</td>\n",
       "      <td>0.4729</td>\n",
       "      <td>0.4974</td>\n",
       "      <td>-1.1900</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.4596</td>\n",
       "      <td>1.8975</td>\n",
       "      <td>1.1120</td>\n",
       "      <td>1.1942</td>\n",
       "      <td>-1.7974</td>\n",
       "      <td>1.1339</td>\n",
       "      <td>0.0259</td>\n",
       "      <td>-0.3529</td>\n",
       "      <td>-1.2327</td>\n",
       "      <td>1.7217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RNF165</td>\n",
       "      <td>494470</td>\n",
       "      <td>-0.4606</td>\n",
       "      <td>0.3564</td>\n",
       "      <td>-0.6800</td>\n",
       "      <td>-1.0563</td>\n",
       "      <td>-0.0377</td>\n",
       "      <td>-0.6829</td>\n",
       "      <td>-0.2854</td>\n",
       "      <td>-0.4336</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0927</td>\n",
       "      <td>0.9103</td>\n",
       "      <td>-0.0023</td>\n",
       "      <td>-0.2898</td>\n",
       "      <td>3.5763</td>\n",
       "      <td>1.3429</td>\n",
       "      <td>0.5726</td>\n",
       "      <td>0.1731</td>\n",
       "      <td>0.5482</td>\n",
       "      <td>1.2239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PHF7</td>\n",
       "      <td>51533</td>\n",
       "      <td>-0.3325</td>\n",
       "      <td>-1.0617</td>\n",
       "      <td>0.2587</td>\n",
       "      <td>-0.2982</td>\n",
       "      <td>-1.2422</td>\n",
       "      <td>0.0558</td>\n",
       "      <td>-0.5011</td>\n",
       "      <td>-0.6418</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0725</td>\n",
       "      <td>0.7219</td>\n",
       "      <td>0.1402</td>\n",
       "      <td>0.8718</td>\n",
       "      <td>-0.9275</td>\n",
       "      <td>-0.0587</td>\n",
       "      <td>0.5240</td>\n",
       "      <td>-0.0311</td>\n",
       "      <td>4.4925</td>\n",
       "      <td>-0.2173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CIDEA</td>\n",
       "      <td>1149</td>\n",
       "      <td>-0.0129</td>\n",
       "      <td>-1.0394</td>\n",
       "      <td>3.2991</td>\n",
       "      <td>-0.2632</td>\n",
       "      <td>-1.0949</td>\n",
       "      <td>1.2628</td>\n",
       "      <td>2.0796</td>\n",
       "      <td>-0.8310</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0679</td>\n",
       "      <td>-0.7126</td>\n",
       "      <td>-0.1523</td>\n",
       "      <td>-0.7593</td>\n",
       "      <td>-0.7141</td>\n",
       "      <td>-0.4324</td>\n",
       "      <td>-0.0336</td>\n",
       "      <td>-0.4003</td>\n",
       "      <td>2.4698</td>\n",
       "      <td>-0.7268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TENT2</td>\n",
       "      <td>167153</td>\n",
       "      <td>-0.7853</td>\n",
       "      <td>0.0337</td>\n",
       "      <td>-0.6649</td>\n",
       "      <td>2.1640</td>\n",
       "      <td>-0.2031</td>\n",
       "      <td>1.0304</td>\n",
       "      <td>0.6046</td>\n",
       "      <td>-1.7557</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6400</td>\n",
       "      <td>-0.1102</td>\n",
       "      <td>1.2719</td>\n",
       "      <td>0.8178</td>\n",
       "      <td>-1.0301</td>\n",
       "      <td>0.6082</td>\n",
       "      <td>0.5608</td>\n",
       "      <td>2.4222</td>\n",
       "      <td>-3.2853</td>\n",
       "      <td>0.4181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20598</th>\n",
       "      <td>VPS72</td>\n",
       "      <td>6944</td>\n",
       "      <td>-0.2908</td>\n",
       "      <td>0.3443</td>\n",
       "      <td>0.4818</td>\n",
       "      <td>0.2503</td>\n",
       "      <td>-0.1057</td>\n",
       "      <td>-0.1657</td>\n",
       "      <td>-0.4730</td>\n",
       "      <td>1.4719</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.9195</td>\n",
       "      <td>-1.4857</td>\n",
       "      <td>-1.4543</td>\n",
       "      <td>0.3791</td>\n",
       "      <td>-0.3989</td>\n",
       "      <td>-1.5529</td>\n",
       "      <td>-0.6349</td>\n",
       "      <td>-0.8160</td>\n",
       "      <td>-1.0902</td>\n",
       "      <td>-0.2811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20599</th>\n",
       "      <td>CSMD3</td>\n",
       "      <td>114788</td>\n",
       "      <td>-0.5286</td>\n",
       "      <td>-0.4379</td>\n",
       "      <td>6.9258</td>\n",
       "      <td>1.0466</td>\n",
       "      <td>-0.1060</td>\n",
       "      <td>0.3284</td>\n",
       "      <td>0.0993</td>\n",
       "      <td>-0.1987</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3776</td>\n",
       "      <td>-0.6366</td>\n",
       "      <td>-0.0607</td>\n",
       "      <td>-0.0475</td>\n",
       "      <td>0.2231</td>\n",
       "      <td>0.0706</td>\n",
       "      <td>0.1188</td>\n",
       "      <td>-0.3231</td>\n",
       "      <td>-0.1251</td>\n",
       "      <td>-0.4265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20600</th>\n",
       "      <td>CC2D1A</td>\n",
       "      <td>54862</td>\n",
       "      <td>0.0068</td>\n",
       "      <td>-0.7520</td>\n",
       "      <td>0.0519</td>\n",
       "      <td>0.2502</td>\n",
       "      <td>-0.3376</td>\n",
       "      <td>-0.4705</td>\n",
       "      <td>-0.6036</td>\n",
       "      <td>-1.1946</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.5877</td>\n",
       "      <td>-1.1169</td>\n",
       "      <td>-0.5420</td>\n",
       "      <td>0.2947</td>\n",
       "      <td>-0.2800</td>\n",
       "      <td>2.5337</td>\n",
       "      <td>-0.8272</td>\n",
       "      <td>-0.1200</td>\n",
       "      <td>4.2708</td>\n",
       "      <td>-1.0090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20601</th>\n",
       "      <td>IGSF9</td>\n",
       "      <td>57549</td>\n",
       "      <td>0.4053</td>\n",
       "      <td>1.2968</td>\n",
       "      <td>0.7962</td>\n",
       "      <td>-0.1634</td>\n",
       "      <td>-0.2418</td>\n",
       "      <td>-0.2545</td>\n",
       "      <td>-0.9814</td>\n",
       "      <td>1.9240</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.6217</td>\n",
       "      <td>-1.5481</td>\n",
       "      <td>-1.2088</td>\n",
       "      <td>0.4594</td>\n",
       "      <td>0.3821</td>\n",
       "      <td>0.3254</td>\n",
       "      <td>0.8187</td>\n",
       "      <td>-0.5648</td>\n",
       "      <td>0.5931</td>\n",
       "      <td>0.9043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20602</th>\n",
       "      <td>FAM71A</td>\n",
       "      <td>149647</td>\n",
       "      <td>-1.1278</td>\n",
       "      <td>0.0321</td>\n",
       "      <td>0.6608</td>\n",
       "      <td>-0.3571</td>\n",
       "      <td>-1.0923</td>\n",
       "      <td>-1.2238</td>\n",
       "      <td>-0.9668</td>\n",
       "      <td>-1.3778</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.4746</td>\n",
       "      <td>1.4956</td>\n",
       "      <td>2.2203</td>\n",
       "      <td>0.3532</td>\n",
       "      <td>-0.2418</td>\n",
       "      <td>0.5514</td>\n",
       "      <td>-0.9335</td>\n",
       "      <td>0.1794</td>\n",
       "      <td>0.0860</td>\n",
       "      <td>0.4096</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20603 rows × 1982 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Hugo_Symbol  Entrez_Gene_Id  MB-0362  MB-0346  MB-0386  MB-0574  \\\n",
       "0            RERE             473  -0.7139   1.2266  -0.0053  -0.4399   \n",
       "1          RNF165          494470  -0.4606   0.3564  -0.6800  -1.0563   \n",
       "2            PHF7           51533  -0.3325  -1.0617   0.2587  -0.2982   \n",
       "3           CIDEA            1149  -0.0129  -1.0394   3.2991  -0.2632   \n",
       "4           TENT2          167153  -0.7853   0.0337  -0.6649   2.1640   \n",
       "...           ...             ...      ...      ...      ...      ...   \n",
       "20598       VPS72            6944  -0.2908   0.3443   0.4818   0.2503   \n",
       "20599       CSMD3          114788  -0.5286  -0.4379   6.9258   1.0466   \n",
       "20600      CC2D1A           54862   0.0068  -0.7520   0.0519   0.2502   \n",
       "20601       IGSF9           57549   0.4053   1.2968   0.7962  -0.1634   \n",
       "20602      FAM71A          149647  -1.1278   0.0321   0.6608  -0.3571   \n",
       "\n",
       "       MB-0185  MB-0503  MB-0641  MB-0201  ...  MB-6192  MB-4820  MB-5527  \\\n",
       "0      -0.5958   0.4729   0.4974  -1.1900  ...  -0.4596   1.8975   1.1120   \n",
       "1      -0.0377  -0.6829  -0.2854  -0.4336  ...  -1.0927   0.9103  -0.0023   \n",
       "2      -1.2422   0.0558  -0.5011  -0.6418  ...  -0.0725   0.7219   0.1402   \n",
       "3      -1.0949   1.2628   2.0796  -0.8310  ...   0.0679  -0.7126  -0.1523   \n",
       "4      -0.2031   1.0304   0.6046  -1.7557  ...   0.6400  -0.1102   1.2719   \n",
       "...        ...      ...      ...      ...  ...      ...      ...      ...   \n",
       "20598  -0.1057  -0.1657  -0.4730   1.4719  ...  -0.9195  -1.4857  -1.4543   \n",
       "20599  -0.1060   0.3284   0.0993  -0.1987  ...  -0.3776  -0.6366  -0.0607   \n",
       "20600  -0.3376  -0.4705  -0.6036  -1.1946  ...  -0.5877  -1.1169  -0.5420   \n",
       "20601  -0.2418  -0.2545  -0.9814   1.9240  ...  -0.6217  -1.5481  -1.2088   \n",
       "20602  -1.0923  -1.2238  -0.9668  -1.3778  ...  -0.4746   1.4956   2.2203   \n",
       "\n",
       "       MB-5167  MB-5465  MB-5453  MB-5471  MB-5127  MB-4313  MB-4823  \n",
       "0       1.1942  -1.7974   1.1339   0.0259  -0.3529  -1.2327   1.7217  \n",
       "1      -0.2898   3.5763   1.3429   0.5726   0.1731   0.5482   1.2239  \n",
       "2       0.8718  -0.9275  -0.0587   0.5240  -0.0311   4.4925  -0.2173  \n",
       "3      -0.7593  -0.7141  -0.4324  -0.0336  -0.4003   2.4698  -0.7268  \n",
       "4       0.8178  -1.0301   0.6082   0.5608   2.4222  -3.2853   0.4181  \n",
       "...        ...      ...      ...      ...      ...      ...      ...  \n",
       "20598   0.3791  -0.3989  -1.5529  -0.6349  -0.8160  -1.0902  -0.2811  \n",
       "20599  -0.0475   0.2231   0.0706   0.1188  -0.3231  -0.1251  -0.4265  \n",
       "20600   0.2947  -0.2800   2.5337  -0.8272  -0.1200   4.2708  -1.0090  \n",
       "20601   0.4594   0.3821   0.3254   0.8187  -0.5648   0.5931   0.9043  \n",
       "20602   0.3532  -0.2418   0.5514  -0.9335   0.1794   0.0860   0.4096  \n",
       "\n",
       "[20603 rows x 1982 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "4c65ce41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "197"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_rna_duplicates = []\n",
    "for i in rna.Hugo_Symbol.unique():\n",
    "    c = rna[rna['Hugo_Symbol']==i].Hugo_Symbol\n",
    "    if len(c) != 1:\n",
    "        list_rna_duplicates.append(i)\n",
    "len(list_rna_duplicates)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "0ad7a306",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "197"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(lista_duplicati))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d449397e",
   "metadata": {},
   "source": [
    "The **meth** dataframe stores, for every patients (columns), the methylation values of 13188 genes (rows). SPIEGARE MEGLIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d5df16b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>MB-0006</th>\n",
       "      <th>MB-0028</th>\n",
       "      <th>MB-0035</th>\n",
       "      <th>MB-0046</th>\n",
       "      <th>MB-0050</th>\n",
       "      <th>MB-0053</th>\n",
       "      <th>MB-0054</th>\n",
       "      <th>MB-0062</th>\n",
       "      <th>MB-0064</th>\n",
       "      <th>...</th>\n",
       "      <th>MB-7279</th>\n",
       "      <th>MB-7281</th>\n",
       "      <th>MB-7283</th>\n",
       "      <th>MB-7285</th>\n",
       "      <th>MB-7288</th>\n",
       "      <th>MB-7289</th>\n",
       "      <th>MB-7291</th>\n",
       "      <th>MB-7292</th>\n",
       "      <th>MB-7293</th>\n",
       "      <th>MB-7296</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2M</td>\n",
       "      <td>0.045031</td>\n",
       "      <td>0.066532</td>\n",
       "      <td>0.015487</td>\n",
       "      <td>0.102439</td>\n",
       "      <td>0.001905</td>\n",
       "      <td>0.039106</td>\n",
       "      <td>0.080311</td>\n",
       "      <td>0.007585</td>\n",
       "      <td>0.003140</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.108374</td>\n",
       "      <td>0.001916</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>0.004049</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.177112</td>\n",
       "      <td>0.015464</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>0.017582</td>\n",
       "      <td>0.038095</td>\n",
       "      <td>0.033333</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009804</td>\n",
       "      <td>0.236311</td>\n",
       "      <td>0.023460</td>\n",
       "      <td>0.001227</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.026393</td>\n",
       "      <td>0.202151</td>\n",
       "      <td>0.061224</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.062745</td>\n",
       "      <td>0.039773</td>\n",
       "      <td>0.007752</td>\n",
       "      <td>0.011628</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003817</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.013043</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.011811</td>\n",
       "      <td>0.003401</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.014151</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AACS</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.019802</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.032967</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012195</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.026316</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADACL2</td>\n",
       "      <td>0.870370</td>\n",
       "      <td>0.958763</td>\n",
       "      <td>0.987395</td>\n",
       "      <td>0.965517</td>\n",
       "      <td>0.969231</td>\n",
       "      <td>0.763359</td>\n",
       "      <td>0.979310</td>\n",
       "      <td>0.958904</td>\n",
       "      <td>0.845588</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.922414</td>\n",
       "      <td>0.962264</td>\n",
       "      <td>0.935484</td>\n",
       "      <td>0.812500</td>\n",
       "      <td>0.939163</td>\n",
       "      <td>0.921348</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.937500</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13183</th>\n",
       "      <td>ZWINT</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.011494</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.016000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13184</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002331</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006061</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.008403</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012500</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13185</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.012658</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.038961</td>\n",
       "      <td>0.014085</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.011173</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13186</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.013445</td>\n",
       "      <td>0.008097</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003155</td>\n",
       "      <td>0.007194</td>\n",
       "      <td>0.005703</td>\n",
       "      <td>0.012384</td>\n",
       "      <td>...</td>\n",
       "      <td>0.015873</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.021028</td>\n",
       "      <td>0.026846</td>\n",
       "      <td>0.001546</td>\n",
       "      <td>0.005533</td>\n",
       "      <td>0.008869</td>\n",
       "      <td>0.013699</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13187</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.002770</td>\n",
       "      <td>0.004566</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003279</td>\n",
       "      <td>0.010381</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002114</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002825</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.012821</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13188 rows × 1419 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Hugo_Symbol   MB-0006   MB-0028   MB-0035   MB-0046   MB-0050   MB-0053  \\\n",
       "0             A2M  0.045031  0.066532  0.015487  0.102439  0.001905  0.039106   \n",
       "1          A4GALT  0.017582  0.038095  0.033333  0.000000  0.009804  0.236311   \n",
       "2            AAAS  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "3            AACS  0.000000  0.019802  0.000000  0.000000  0.000000  0.000000   \n",
       "4         AADACL2  0.870370  0.958763  0.987395  0.965517  0.969231  0.763359   \n",
       "...           ...       ...       ...       ...       ...       ...       ...   \n",
       "13183       ZWINT  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "13184        ZXDC  0.000000  0.000000  0.000000  0.002331  0.000000  0.000000   \n",
       "13185      ZYG11B  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "13186         ZYX  0.013445  0.008097  0.000000  0.000000  0.000000  0.003155   \n",
       "13187       ZZEF1  0.002770  0.004566  0.000000  0.000000  0.000000  0.003279   \n",
       "\n",
       "        MB-0054   MB-0062   MB-0064  ...   MB-7279   MB-7281   MB-7283  \\\n",
       "0      0.080311  0.007585  0.003140  ...  0.000000  0.108374  0.001916   \n",
       "1      0.023460  0.001227  0.004673  ...  0.000000  0.026393  0.202151   \n",
       "2      0.000000  0.003817  0.000000  ...  0.000000  0.000000  0.013043   \n",
       "3      0.032967  0.000000  0.012195  ...  0.000000  0.000000  0.026316   \n",
       "4      0.979310  0.958904  0.845588  ...       NaN  0.922414  0.962264   \n",
       "...         ...       ...       ...  ...       ...       ...       ...   \n",
       "13183  0.000000  0.000000  0.011494  ...  0.000000  0.000000  0.000000   \n",
       "13184  0.006061  0.000000  0.008403  ...  0.000000  0.000000  0.012500   \n",
       "13185  0.012658  0.000000  0.000000  ...       NaN  0.000000  0.038961   \n",
       "13186  0.007194  0.005703  0.012384  ...  0.015873  0.000000  0.021028   \n",
       "13187  0.010381  0.000000  0.000000  ...  0.000000  0.002114  0.000000   \n",
       "\n",
       "        MB-7285   MB-7288   MB-7289   MB-7291   MB-7292   MB-7293  MB-7296  \n",
       "0      0.004673  0.004049  0.000000  0.177112  0.015464  0.090909      0.1  \n",
       "1      0.061224  0.000000  0.062745  0.039773  0.007752  0.011628      0.0  \n",
       "2      0.000000  0.000000  0.011811  0.003401  0.000000  0.014151      0.0  \n",
       "3           NaN  0.000000  0.000000  0.007194  0.000000  0.000000      0.0  \n",
       "4      0.935484  0.812500  0.939163  0.921348       NaN  0.937500      1.0  \n",
       "...         ...       ...       ...       ...       ...       ...      ...  \n",
       "13183  0.000000  0.000000  0.000000  0.016000  0.000000  0.000000      0.0  \n",
       "13184  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000      0.0  \n",
       "13185  0.014085  0.000000  0.011173  0.000000       NaN  0.000000      0.0  \n",
       "13186  0.026846  0.001546  0.005533  0.008869  0.013699  0.000000      0.0  \n",
       "13187  0.000000  0.000000  0.000000  0.002825       NaN  0.012821      0.0  \n",
       "\n",
       "[13188 rows x 1419 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e300f89",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b302b306",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3f18324",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65c95a10",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "826179a8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48b611c8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "abb30f68",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "72fd56ca",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "591d683b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e01a4534",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ce7abf7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "raw",
   "id": "5635fb04",
   "metadata": {},
   "source": [
    "Copy-number alterations from DNA copy. Calls made after normal contamination correction and CNV removal using thresholds. Values: -2 = homozygous deletion; -1 = hemizygous deletion; 0 = neutral / no change; 1 = gain; 2 = high level amplification."
   ]
  },
  {
   "cell_type": "raw",
   "id": "bb738253",
   "metadata": {},
   "source": [
    "NPI\n",
    "The Nottingham prognostic index (NPI) is used to determine prognosis following surgery for breast cancer.[1][2] Its value is calculated using three pathological criteria: the size of the tumour; the number of involved lymph nodes; and the grade of the tumour.[1] It is calculated to select patients for adjuvant treatment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6705fa2",
   "metadata": {},
   "outputs": [],
   "source": [
    "CELLULARITY\n",
    "tumor content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74f01bcc",
   "metadata": {},
   "outputs": [],
   "source": [
    "useless_cols = []\n",
    "for i in mut.columns:\n",
    "    col = mut[i].unique()\n",
    "    if len(col) == 1:\n",
    "        useless_cols.append(i)\n",
    "\n",
    "new_cols = []\n",
    "for i in mut.columns:\n",
    "    if i not in useless_cols:\n",
    "        new_cols.append(i)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1159bd04",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "482b464d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2ede310",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7da2332d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93e69f82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "6033a1c6",
   "metadata": {},
   "source": [
    "## Genetic alterations in breast cancer <a class=\"anchor\" id=\"2\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "9b1503a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene1 = pd.read_csv('data/gene_list/Census_allSun_Jan_14_17_21_06_2024.csv')\n",
    "gene2 = pd.read_csv('data/gene_list/Familial_breast_cancer.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "dc2575f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Name</th>\n",
       "      <th>Entrez GeneId</th>\n",
       "      <th>Genome Location</th>\n",
       "      <th>Tier</th>\n",
       "      <th>Hallmark</th>\n",
       "      <th>Chr Band</th>\n",
       "      <th>Somatic</th>\n",
       "      <th>Germline</th>\n",
       "      <th>Tumour Types(Somatic)</th>\n",
       "      <th>Tumour Types(Germline)</th>\n",
       "      <th>Cancer Syndrome</th>\n",
       "      <th>Tissue Type</th>\n",
       "      <th>Molecular Genetics</th>\n",
       "      <th>Role in Cancer</th>\n",
       "      <th>Mutation Types</th>\n",
       "      <th>Translocation Partner</th>\n",
       "      <th>Other Germline Mut</th>\n",
       "      <th>Other Syndrome</th>\n",
       "      <th>Synonyms</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>10:50799421-50885675</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10q11.23</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>oncogene</td>\n",
       "      <td>Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACF,ACF64,ACF65,APOBEC1CF,ASP,CCDS73133.1,ENSG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABI1</td>\n",
       "      <td>abl-interactor 1</td>\n",
       "      <td>10006.0</td>\n",
       "      <td>10:26746593-26860935</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>10p12.1</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>TSG, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>KMT2A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABI-1,CCDS7150.1,E3B1,ENSG00000136754.17,NM_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>v-abl Abelson murine leukemia viral oncogene h...</td>\n",
       "      <td>25.0</td>\n",
       "      <td>9:130713946-130885683</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>9q34.12</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CML, ALL, T-ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T, Mis</td>\n",
       "      <td>BCR, ETV6, NUP214</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABL,CCDS35165.1,ENSG00000097007.17,JTK7,NM_007...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>c-abl oncogene 2, non-receptor tyrosine kinase</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1:179099327-179229601</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1q25.2</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>ETV6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABLL,ARG,CCDS30947.1,ENSG00000143322.19,NM_007...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACKR3</td>\n",
       "      <td>atypical chemokine receptor 3</td>\n",
       "      <td>57007.0</td>\n",
       "      <td>2:236569641-236582358</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>2q37.3</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lipoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>HMGA2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS2516.1,CMKOR1,CXCR7,ENSG00000144476.5,GPR1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>738</th>\n",
       "      <td>ZNF429</td>\n",
       "      <td>zinc finger protein 429</td>\n",
       "      <td>353088.0</td>\n",
       "      <td>19:21505564-21538078</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19p12</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GBM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS42537.1,ENSG00000197013.9,NM_001001415.2,N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>739</th>\n",
       "      <td>ZNF479</td>\n",
       "      <td>zinc finger protein 479</td>\n",
       "      <td>90827.0</td>\n",
       "      <td>7:57119614-57139864</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7p11.2</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lung cancer, bladder carcinoma, prostate carci...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS43590.1,ENSG00000185177.12,KR19,NM_033273....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>740</th>\n",
       "      <td>ZNF521</td>\n",
       "      <td>zinc finger protein 521</td>\n",
       "      <td>25925.0</td>\n",
       "      <td>18:25061926-25352152</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>18q11.2</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS32806.1,EHZF,ENSG00000198795.10,Evi3,NM_01...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>741</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>zinc and ring finger 3</td>\n",
       "      <td>84133.0</td>\n",
       "      <td>22:28883592-29057487</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22q12.1</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colorectal cancer, adrenocortical carcinoma, g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TSG</td>\n",
       "      <td>N, F, Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK747E2.3,CCDS56225.1,ENSG00000183579.15,FLJ22...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>742</th>\n",
       "      <td>ZRSR2</td>\n",
       "      <td>zinc finger (CCCH type), RNA-binding motif and...</td>\n",
       "      <td>8233.0</td>\n",
       "      <td>X:15790472-15823260</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Xp22.2</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MDS, CLL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Rec</td>\n",
       "      <td>TSG</td>\n",
       "      <td>F, S, Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS14172.1,ENSG00000169249.12,NM_005089.3,NP_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>743 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Symbol                                               Name  \\\n",
       "0          A1CF                     APOBEC1 complementation factor   \n",
       "1          ABI1                                   abl-interactor 1   \n",
       "2          ABL1  v-abl Abelson murine leukemia viral oncogene h...   \n",
       "3          ABL2     c-abl oncogene 2, non-receptor tyrosine kinase   \n",
       "4         ACKR3                      atypical chemokine receptor 3   \n",
       "..          ...                                                ...   \n",
       "738      ZNF429                            zinc finger protein 429   \n",
       "739      ZNF479                            zinc finger protein 479   \n",
       "740      ZNF521                            zinc finger protein 521   \n",
       "741       ZNRF3                             zinc and ring finger 3   \n",
       "742       ZRSR2  zinc finger (CCCH type), RNA-binding motif and...   \n",
       "\n",
       "     Entrez GeneId        Genome Location  Tier Hallmark  Chr Band Somatic  \\\n",
       "0          29974.0   10:50799421-50885675     2      NaN  10q11.23     yes   \n",
       "1          10006.0   10:26746593-26860935     1      Yes   10p12.1     yes   \n",
       "2             25.0  9:130713946-130885683     1      Yes   9q34.12     yes   \n",
       "3             27.0  1:179099327-179229601     1      NaN    1q25.2     yes   \n",
       "4          57007.0  2:236569641-236582358     1      Yes    2q37.3     yes   \n",
       "..             ...                    ...   ...      ...       ...     ...   \n",
       "738       353088.0   19:21505564-21538078     2      NaN     19p12     yes   \n",
       "739        90827.0    7:57119614-57139864     2      NaN    7p11.2     yes   \n",
       "740        25925.0   18:25061926-25352152     1      NaN   18q11.2     yes   \n",
       "741        84133.0   22:28883592-29057487     2      NaN   22q12.1     yes   \n",
       "742         8233.0    X:15790472-15823260     1      Yes    Xp22.2     yes   \n",
       "\n",
       "    Germline                              Tumour Types(Somatic)  \\\n",
       "0        NaN                                           melanoma   \n",
       "1        NaN                                                AML   \n",
       "2        NaN                                    CML, ALL, T-ALL   \n",
       "3        NaN                                                AML   \n",
       "4        NaN                                             lipoma   \n",
       "..       ...                                                ...   \n",
       "738      NaN                                                GBM   \n",
       "739      NaN  lung cancer, bladder carcinoma, prostate carci...   \n",
       "740      NaN                                                ALL   \n",
       "741      NaN  colorectal cancer, adrenocortical carcinoma, g...   \n",
       "742      NaN                                           MDS, CLL   \n",
       "\n",
       "    Tumour Types(Germline) Cancer Syndrome Tissue Type Molecular Genetics  \\\n",
       "0                      NaN             NaN           E                NaN   \n",
       "1                      NaN             NaN           L                Dom   \n",
       "2                      NaN             NaN           L                Dom   \n",
       "3                      NaN             NaN           L                Dom   \n",
       "4                      NaN             NaN           M                Dom   \n",
       "..                     ...             ...         ...                ...   \n",
       "738                    NaN             NaN           O                NaN   \n",
       "739                    NaN             NaN           E                NaN   \n",
       "740                    NaN             NaN           L                Dom   \n",
       "741                    NaN             NaN           E                NaN   \n",
       "742                    NaN             NaN           L                Rec   \n",
       "\n",
       "       Role in Cancer Mutation Types Translocation Partner Other Germline Mut  \\\n",
       "0            oncogene            Mis                   NaN                NaN   \n",
       "1         TSG, fusion              T                 KMT2A                NaN   \n",
       "2    oncogene, fusion         T, Mis     BCR, ETV6, NUP214                NaN   \n",
       "3    oncogene, fusion              T                  ETV6                NaN   \n",
       "4    oncogene, fusion              T                 HMGA2                NaN   \n",
       "..                ...            ...                   ...                ...   \n",
       "738               NaN            Mis                   NaN                NaN   \n",
       "739               NaN            Mis                   NaN                NaN   \n",
       "740  oncogene, fusion              T                  PAX5                NaN   \n",
       "741               TSG      N, F, Mis                   NaN                NaN   \n",
       "742               TSG      F, S, Mis                   NaN                NaN   \n",
       "\n",
       "    Other Syndrome                                           Synonyms  \n",
       "0              NaN  ACF,ACF64,ACF65,APOBEC1CF,ASP,CCDS73133.1,ENSG...  \n",
       "1              NaN  ABI-1,CCDS7150.1,E3B1,ENSG00000136754.17,NM_00...  \n",
       "2              NaN  ABL,CCDS35165.1,ENSG00000097007.17,JTK7,NM_007...  \n",
       "3              NaN  ABLL,ARG,CCDS30947.1,ENSG00000143322.19,NM_007...  \n",
       "4              NaN  CCDS2516.1,CMKOR1,CXCR7,ENSG00000144476.5,GPR1...  \n",
       "..             ...                                                ...  \n",
       "738            NaN  CCDS42537.1,ENSG00000197013.9,NM_001001415.2,N...  \n",
       "739            NaN  CCDS43590.1,ENSG00000185177.12,KR19,NM_033273....  \n",
       "740            NaN  CCDS32806.1,EHZF,ENSG00000198795.10,Evi3,NM_01...  \n",
       "741            NaN  BK747E2.3,CCDS56225.1,ENSG00000183579.15,FLJ22...  \n",
       "742            NaN  CCDS14172.1,ENSG00000169249.12,NM_005089.3,NP_...  \n",
       "\n",
       "[743 rows x 20 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3042b387",
   "metadata": {},
   "source": [
    "Select genes that are related to breast cancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "3ffa09d4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene1 list\n",
      "Number of labeled genes:  700\n",
      "Number of labeled breast cancer related genes:  42\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['AKT1',\n",
       " 'ARID1A',\n",
       " 'ARID1B',\n",
       " 'ASPM',\n",
       " 'BAP1',\n",
       " 'BARD1',\n",
       " 'BRCA2',\n",
       " 'CASP8',\n",
       " 'CCND1',\n",
       " 'CDH1',\n",
       " 'CDKN1B',\n",
       " 'CTCF',\n",
       " 'EP300',\n",
       " 'ERBB2',\n",
       " 'ESR1',\n",
       " 'ETV6',\n",
       " 'FADD',\n",
       " 'FBLN2',\n",
       " 'FLNA',\n",
       " 'FOXA1',\n",
       " 'GATA3',\n",
       " 'GOLPH3',\n",
       " 'HGF',\n",
       " 'IKZF3',\n",
       " 'IRS4',\n",
       " 'KEAP1',\n",
       " 'MAP2K4',\n",
       " 'MAP3K1',\n",
       " 'MAP3K13',\n",
       " 'NCOR1',\n",
       " 'NOTCH1',\n",
       " 'NTRK3',\n",
       " 'PBRM1',\n",
       " 'PIK3CA',\n",
       " 'PPM1D',\n",
       " 'RAD50',\n",
       " 'RB1',\n",
       " 'SALL4',\n",
       " 'SMARCD1',\n",
       " 'TBX3',\n",
       " 'TP53',\n",
       " 'ZMYM3']"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print('gene1 list')\n",
    "gene1_red = gene1[['Gene Symbol','Tumour Types(Somatic)']].dropna().copy()\n",
    "gene1_red.index = np.arange(len(gene1_red))\n",
    "print('Number of labeled genes: ',len(gene1_red))\n",
    "\n",
    "breast_index = []\n",
    "for i in range(len(gene1_red)):\n",
    "    label = gene1_red['Tumour Types(Somatic)'].iloc[i]\n",
    "    if ('breast' in label):\n",
    "        breast_index = breast_index + [gene1_red.iloc[i].name]\n",
    "       \n",
    "gene1_red = gene1_red.iloc[breast_index]\n",
    "print('Number of labeled breast cancer related genes: ',len(gene1_red))\n",
    "\n",
    "gene_list1 = gene1_red['Gene Symbol'].tolist()\n",
    "gene_list1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "89f29efc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene2 list\n",
      "Number of breast cancer related genes:  27\n"
     ]
    }
   ],
   "source": [
    "gene_list2 = gene2['Gene Symbol'].tolist()\n",
    "print('gene2 list')\n",
    "print('Number of breast cancer related genes: ',len(gene_list2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1d8b4233",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of breast cancer related genes:  63\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['TP53',\n",
       " 'MSH2',\n",
       " 'PALB2',\n",
       " 'PBRM1',\n",
       " 'NOTCH1',\n",
       " 'CDH1',\n",
       " 'ARID1B',\n",
       " 'RAD54L',\n",
       " 'PTEN',\n",
       " 'CHEK2',\n",
       " 'ARID1A',\n",
       " 'BRCA2',\n",
       " 'MLH1',\n",
       " 'HGF',\n",
       " 'BAP1',\n",
       " 'ZMYM3',\n",
       " 'RAD51C',\n",
       " 'NBN',\n",
       " 'ESR1',\n",
       " 'AR',\n",
       " 'ATRIP',\n",
       " 'MSH6',\n",
       " 'FLNA',\n",
       " 'GOLPH3',\n",
       " 'RB1',\n",
       " 'SALL4',\n",
       " 'ATM',\n",
       " 'ERBB2',\n",
       " 'CTCF',\n",
       " 'CCND1',\n",
       " 'MAP3K1',\n",
       " 'RAD50',\n",
       " 'ASPM',\n",
       " 'CDKN1B',\n",
       " 'BARD1',\n",
       " 'RAD51D',\n",
       " 'FADD',\n",
       " 'XRCC2',\n",
       " 'MAP3K13',\n",
       " 'BRIP1',\n",
       " 'KEAP1',\n",
       " 'GATA3',\n",
       " 'PPM1D',\n",
       " 'BRCA1',\n",
       " 'FOXA1',\n",
       " 'MAP2K4',\n",
       " 'MUTYH',\n",
       " 'AKT1',\n",
       " 'SMARCD1',\n",
       " 'NCOR1',\n",
       " 'NTRK3',\n",
       " 'STK11',\n",
       " 'EP300',\n",
       " 'PIK3CA',\n",
       " 'EPCAM',\n",
       " 'IKZF3',\n",
       " 'RRAS2',\n",
       " 'PMS2',\n",
       " 'FBLN2',\n",
       " 'CASP8',\n",
       " 'IRS4',\n",
       " 'TBX3',\n",
       " 'ETV6']"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_list = gene_list1 + gene_list2\n",
    "gene_list = list(set(gene_list))\n",
    "print('Total number of breast cancer related genes: ',len(gene_list))\n",
    "gene_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "f9826701",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(r'data/gene_list.txt', 'w') as file:\n",
    "    for item in gene_list:\n",
    "        file.write(item+'\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f06e3b3c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e25921ef",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8284297",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "eabdc332",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['TP53', 'MLLT4', 'NF2', 'SF3B1', 'NT5E', 'EGFR', 'PRKCQ', 'AHNAK',\n",
       "       'CLK3', 'ERBB2', 'MUC16', 'EP300', 'HIST1H2BC', 'UBR5', 'USP28',\n",
       "       'AHNAK2', 'BIRC6', 'PIK3CA', 'MAP3K13', 'COL12A1', 'FOXO3',\n",
       "       'SYNE1', 'NRG3', 'DNAH2', 'NF1', 'SETDB1', 'COL6A3', 'ATR',\n",
       "       'KMT2C', 'GPR124', 'ARID2', 'ERBB3', 'SIK1', 'TAF1', 'PTEN',\n",
       "       'AKT1', 'FANCA', 'MLL2', 'BRCA1', 'DNAH11', 'AKAP9', 'STAB2',\n",
       "       'HERC2', 'CACNA2D3', 'RYR2', 'PALLD', 'ARID1A', 'LAMB3', 'SBNO1',\n",
       "       'NCOA3', 'STK11', 'MAP3K1', 'PIK3R1', 'UTRN', 'TG', 'HRAS',\n",
       "       'FANCD2', 'SHANK2', 'NPNT', 'USH2A', 'PBRM1', 'CDH1', 'MYH9',\n",
       "       'MAP2K4', 'NCOR1', 'PTPRD', 'TBX3', 'CBFB', 'CTCF', 'RUNX1',\n",
       "       'COL22A1', 'NCOR2', 'DNAH5', 'GATA3', 'AFF2', 'ALK', 'MAP3K10',\n",
       "       'BRCA2', 'KDM3A', 'FAM20C', 'MAGEA8', 'L1CAM', 'PRKCZ', 'GPS2',\n",
       "       'USP9X', 'PTPN22', 'PDE4DIP', 'GPR32', 'LARGE', 'THSD7A', 'SIAH1',\n",
       "       'LAMA2', 'HDAC9', 'ARID1B', 'FLT3', 'PRKCE', 'AGTR2', 'PTPRM',\n",
       "       'CTNNA3', 'CTNNA1', 'ROS1', 'RPGR', 'PRR16', 'ASXL1', 'SETD1A',\n",
       "       'AKT2', 'MBL2', 'GH1', 'SGCD', 'KDM6A', 'BRAF', 'CASP8', 'SETD2',\n",
       "       'SMARCC1', 'NR3C1', 'DCAF4L2', 'MYO1A', 'CHEK2', 'KRAS', 'LIFR',\n",
       "       'DTWD2', 'NOTCH1', 'GLDC', 'THADA', 'PRPS2', 'FRMD3', 'MYO3A',\n",
       "       'ACVRL1', 'SMARCC2', 'TTYH1', 'FOXO1', 'OR6A2', 'APC', 'RB1',\n",
       "       'SMARCB1', 'CLRN2', 'MTAP', 'FBXW7', 'ARID5B', 'PRKACG', 'PRKG1',\n",
       "       'BCAS3', 'ZFP36L1', 'TBL1XR1', 'BAP1', 'SPACA1', 'JAK1', 'CHD1',\n",
       "       'FOXP1', 'ASXL2', 'AGMO', 'LIPI', 'SMAD4', 'BRIP1', 'ERBB4',\n",
       "       'MEN1', 'CDKN1B', 'CDKN2A', 'RASGEF1B', 'NDFIP1', 'PPP2R2A',\n",
       "       'SMAD2', 'SIK2', 'NEK1', 'SMARCD1', 'NR2F1', 'KLRG1', 'TAF4B',\n",
       "       'LDLRAP1', 'CCND3', 'PPP2CB', 'STMN2', 'NRAS'], dtype=object)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut.Hugo_Symbol.unique()#.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c614919f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5fb443ae",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "default_env_kernel",
   "language": "python",
   "name": "default_env_kernel"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
